{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "22a2a50d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "import itertools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "b64e3767",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_in_fasta(fasta_path):\n",
    "    seqs = []\n",
    "\n",
    "    for record in SeqIO.parse(fasta_path, \"fasta\"):\n",
    "\n",
    "        seqs.append([record.id, record.seq])\n",
    "\n",
    "    seqs = pd.DataFrame(seqs, columns=['id', 'seq'])\n",
    "    \n",
    "    return seqs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "af837f6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_df_to_fasta(df, seq_col, name_col, fasta_out_path):\n",
    "    \n",
    "    sequences = []\n",
    "\n",
    "    for i in df.index:\n",
    "        \n",
    "        seq_id = df.loc[i, name_col].replace(\" \", \"_\")\n",
    "    \n",
    "        sequences.append(SeqRecord(Seq(df.loc[i, seq_col]), id=seq_id, description=\"\"))\n",
    "\n",
    "    with open(fasta_out_path, \"w\") as output_handle:\n",
    "        SeqIO.write(sequences, output_handle, \"fasta\")\n",
    "        \n",
    "    return\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95b8183b",
   "metadata": {},
   "source": [
    "## quick investigation of coordinates of protein features:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "b1c3858f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # ecoli = SeqIO.read(\"motif/ecoli_biob_aa.fa\", \"fasta\")\n",
    "# # mtb = SeqIO.read(\"bioB_h37rv_protein.fa\", \"fasta\")\n",
    "\n",
    "# nt = SeqIO.read(\"bspA_h37rv.fa\", \"fasta\")\n",
    "# aa = SeqIO.read(\"bspA_h37rv_protein.fa\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "5312da0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# nt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "d994de69",
   "metadata": {},
   "outputs": [],
   "source": [
    "# nt[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "85b26b6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# (49*3)+3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f2cde854",
   "metadata": {},
   "outputs": [],
   "source": [
    "# nt[126:147]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "644b231f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# nt[126:147].translate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "973b9f20",
   "metadata": {},
   "outputs": [],
   "source": [
    "# aa[42:49]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "a4421292",
   "metadata": {},
   "outputs": [],
   "source": [
    "# aa_cons = \"------------------------------------------------------------YN-YTG-----V---G---A---------AARLG--LEPPRFC-QCGRRMVVQV-PDGW-ARCSRHG-VDS--L--R--------\"\n",
    "# nt_cons = \"----------------------------------ATGGTGG---------------------------------------AAATCGTGGCTGGAAAACAACGCGCTCC------------------------GGTCGCTGC-----------------CGG----------CGTGTACAACGTGTACACCGGG--------------------------------------------------------------------GAACTGGCGGATACGGCCACGCCGACAGCG-----GCTCG-----GATGG---------GTC--TGGAGCCCCCCCGGTTCTGTGCGCAGTGCGGTCGCCGGATGGTCGTCCAGGTCCGGCCCGACGGCTGGTGGGCGCGCTGTTCTCGCCACGGGC---AGGTGGACTCGG---------------------CCGACTTGGCGACACAGCGGTGA---------\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "3bbd398a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# aa_cons_expand = \"\"\n",
    "# for i in aa_cons:\n",
    "#     aa_cons_expand = aa_cons_expand + \" \" + i + \" \""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "ac5e87c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# aa_cons_expand\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "622e2760",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "435"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# len(aa_cons_expand)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "888b3b05",
   "metadata": {},
   "outputs": [],
   "source": [
    "# nt_cons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "626ffc78",
   "metadata": {},
   "outputs": [],
   "source": [
    "# mtb[158:168]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "30f0c169",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ecoli[127]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a6454feb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ecoli[123:130]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "21f68a82",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ecoli[187]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "32bc441e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ecoli[184:192]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "cb86da29",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ecoli[90:100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f61962d9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "ce32eaac",
   "metadata": {},
   "source": [
    "### write reference taxids to list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4cfa739a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"motif/actinomycetota_bvbrc_reference_genomes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8c8a6e39",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[['NCBI Taxon ID']].to_csv(\"atinomycetota_bvbrc_reference_taxid_list.csv\", index=None, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6de87767",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "bce44126",
   "metadata": {},
   "source": [
    "### Read in bsap and biob blast hits against reference genomes; divide out bsap independent and bsap dependent biob copies:\n",
    "- Filtered hits to include only hits with at least 33% the length of the biob/bsap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "d00a1a8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_cols = ['seqid', 'evalue', 'bitscore', 'pident', 'length', 'staxid', 'ssname', 'scomname', 'sstart', 'ssend', 'sseq']\n",
    "\n",
    "bsap_all = pd.read_csv(\"motif_prev/bsap_bvbrc_blastn_hits_31Aug2023.tsv\", delimiter=\"\\t\", names=blast_cols, index_col=False)\n",
    "biob = pd.read_csv(\"motif_prev/biob_bvbrc_blastn_hits_31Aug2023.tsv\", delimiter=\"\\t\", names=blast_cols, index_col=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "a7180694",
   "metadata": {},
   "outputs": [],
   "source": [
    "# biob = biob[biob['length'] > 349].reset_index(drop=True)\n",
    "bsap = bsap_all[bsap_all['length'] > 79].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "e2e3c446",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob = biob.drop_duplicates(subset='staxid')\n",
    "bsap = bsap.drop_duplicates(subset='staxid')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "029752ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "bsap_phylogeny_tax_list = pd.read_csv(\"final_nt_workflow/bsap_nt_subset50_rooted_species_order.txt\", header=None, index_col=False\n",
    "                                     ).transpose()[0].str.replace(\"_\", \" \").tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "f7e65d17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "261.75"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(349*3)/4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "5c974b1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59.25"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(79*3)/4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "558b9a51",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_ind_species_list = biob[~biob['staxid'].isin(bsap['staxid'].unique())]['scomname']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "17d069b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2798"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(biob_ind_species_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "f415d40d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "25"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# unique genuses in bsap phylogeny analysis:\n",
    "len(pd.Series(bsap_phylogeny_tax_list).str.split(\" \", expand=True)[0].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "62415f83",
   "metadata": {},
   "outputs": [],
   "source": [
    "## allowing that there are some myco derivative species showing up in the bsap independent biob class; we're still going to exclude those\n",
    "## and any species that sit too close phylogenetically to our bsap investigation --\n",
    "## definetely going to keep track of these species that are dropped out and run a motif based analysis on them later on, to check in on the\n",
    "## appearance of biob function. But generally these look like strains with uncertain taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "23d5cd4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(511, 11)"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bsap.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "78a4716c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3309, 11)"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "af8518c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(biob[~biob['staxid'].isin(bsap['staxid'].unique())]['staxid'].unique()\n",
    "            ).to_csv(\"motif/putative_biob_independent_staxid_list.csv\", index=False, header=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f960e36a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "a177d433",
   "metadata": {},
   "outputs": [],
   "source": [
    "# biob_subset = biob[~biob['staxid'].isin(bsap['staxid'].unique()) & \n",
    "#                    ~(biob['ssname'].str.contains('Myco')) &\n",
    "#                    ~(biob['ssname'].str.contains(\"Corynebacterium\"))\n",
    "#     ].reset_index(drop=True).copy() #.to_csv('motif/putative_biob_independent.csv', index=False)\n",
    "\n",
    "biob_subset = biob[~biob['staxid'].isin(bsap['staxid'].unique())].reset_index(drop=True).copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "edd72ff3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>sseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gi|1888951912|ref|NZ_JACGWZ010000001.1|</td>\n",
       "      <td>1.730000e-05</td>\n",
       "      <td>48.2</td>\n",
       "      <td>76.543</td>\n",
       "      <td>81</td>\n",
       "      <td>1000566</td>\n",
       "      <td>Halosaccharopolyspora lacisalsi</td>\n",
       "      <td>Halosaccharopolyspora lacisalsi</td>\n",
       "      <td>1356959</td>\n",
       "      <td>1357038</td>\n",
       "      <td>TCGTCACCAACCGGACGGCCTTCTCGCCGGCGACCGCGGAAGCCGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gi|1890800121|ref|NZ_JACHBQ010000001.1|</td>\n",
       "      <td>2.000000e-03</td>\n",
       "      <td>41.9</td>\n",
       "      <td>96.000</td>\n",
       "      <td>25</td>\n",
       "      <td>1001240</td>\n",
       "      <td>Cryobacterium roopkundense</td>\n",
       "      <td>Cryobacterium roopkundense</td>\n",
       "      <td>976391</td>\n",
       "      <td>976415</td>\n",
       "      <td>GCCTGAACCAGGACCAGGTTCTGGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gi|1124504080|ref|NZ_FOJD01000002.1|</td>\n",
       "      <td>2.100000e-02</td>\n",
       "      <td>37.4</td>\n",
       "      <td>87.500</td>\n",
       "      <td>32</td>\n",
       "      <td>100225</td>\n",
       "      <td>Austwickia chelonae</td>\n",
       "      <td>Austwickia chelonae</td>\n",
       "      <td>539733</td>\n",
       "      <td>539763</td>\n",
       "      <td>CCGAGTTCCACCTCGTGGCCGCGG-GCGCGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gi|1441077360|ref|NZ_QQZY01000004.1|</td>\n",
       "      <td>1.550000e-09</td>\n",
       "      <td>60.8</td>\n",
       "      <td>69.484</td>\n",
       "      <td>213</td>\n",
       "      <td>1002870</td>\n",
       "      <td>Gaiella occulta</td>\n",
       "      <td>Gaiella occulta</td>\n",
       "      <td>267788</td>\n",
       "      <td>267995</td>\n",
       "      <td>CTGCACGCTCGGCATGCTCACGGCCGAGCAAGCGGAGGCGCTCGC-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gi|357407371|ref|NC_016113.1|</td>\n",
       "      <td>9.110000e-05</td>\n",
       "      <td>47.3</td>\n",
       "      <td>75.294</td>\n",
       "      <td>85</td>\n",
       "      <td>1003195</td>\n",
       "      <td>Streptantibioticus cattleyicolor NRRL 8057 = D...</td>\n",
       "      <td>Streptantibioticus cattleyicolor NRRL 8057 = D...</td>\n",
       "      <td>1386387</td>\n",
       "      <td>1386469</td>\n",
       "      <td>CGGCGAGGCGGGCGCGGACGCCTACAACCACAACCTCAACACCGCC...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     seqid        evalue  bitscore  pident  \\\n",
       "0  gi|1888951912|ref|NZ_JACGWZ010000001.1|  1.730000e-05      48.2  76.543   \n",
       "1  gi|1890800121|ref|NZ_JACHBQ010000001.1|  2.000000e-03      41.9  96.000   \n",
       "2     gi|1124504080|ref|NZ_FOJD01000002.1|  2.100000e-02      37.4  87.500   \n",
       "3     gi|1441077360|ref|NZ_QQZY01000004.1|  1.550000e-09      60.8  69.484   \n",
       "4            gi|357407371|ref|NC_016113.1|  9.110000e-05      47.3  75.294   \n",
       "\n",
       "   length   staxid                                             ssname  \\\n",
       "0      81  1000566                    Halosaccharopolyspora lacisalsi   \n",
       "1      25  1001240                         Cryobacterium roopkundense   \n",
       "2      32   100225                                Austwickia chelonae   \n",
       "3     213  1002870                                    Gaiella occulta   \n",
       "4      85  1003195  Streptantibioticus cattleyicolor NRRL 8057 = D...   \n",
       "\n",
       "                                            scomname   sstart    ssend  \\\n",
       "0                    Halosaccharopolyspora lacisalsi  1356959  1357038   \n",
       "1                         Cryobacterium roopkundense   976391   976415   \n",
       "2                                Austwickia chelonae   539733   539763   \n",
       "3                                    Gaiella occulta   267788   267995   \n",
       "4  Streptantibioticus cattleyicolor NRRL 8057 = D...  1386387  1386469   \n",
       "\n",
       "                                                sseq  \n",
       "0  TCGTCACCAACCGGACGGCCTTCTCGCCGGCGACCGCGGAAGCCGT...  \n",
       "1                          GCCTGAACCAGGACCAGGTTCTGGC  \n",
       "2                   CCGAGTTCCACCTCGTGGCCGCGG-GCGCGGA  \n",
       "3  CTGCACGCTCGGCATGCTCACGGCCGAGCAAGCGGAGGCGCTCGC-...  \n",
       "4  CGGCGAGGCGGGCGCGGACGCCTACAACCACAACCTCAACACCGCC...  "
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "33dae0cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Streptomyces       532\n",
       "Corynebacterium    117\n",
       "Microbacterium     103\n",
       "Nocardioides       100\n",
       "Bifidobacterium     95\n",
       "                  ... \n",
       "Specibacter          1\n",
       "Rubneribacter        1\n",
       "Baekduia             1\n",
       "Peptidiphaga         1\n",
       "Phytomonospora       1\n",
       "Name: 0, Length: 433, dtype: int64"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset['ssname'].str.split(\" \", expand=True)[0].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5e82baa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "8e690c85",
   "metadata": {},
   "outputs": [],
   "source": [
    "# write_df_to_fasta(biob_subset, \"motif/putative1_biob_independent_seqs.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec548946",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "d0d10829",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for i in biob[ (~biob['staxid'].isin(bsap['staxid'].unique())) & \n",
    "#      (biob['scomname'].str.contains(\"Myco\")) ]['seqid'].str.split(\"|\", expand=True)[3].tolist():\n",
    "#     print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "97972047",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>sseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [seqid, evalue, bitscore, pident, length, staxid, ssname, scomname, sstart, ssend, sseq]\n",
       "Index: []"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bsap_all[bsap_all['scomname'].isin(biob[ (~biob['staxid'].isin(bsap['staxid'].unique())) & \n",
    "                                        (biob['scomname'].str.contains(\"Mycobacter\")) ]) ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b375f6c0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b369c6bd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "e3ab23f3",
   "metadata": {},
   "source": [
    "## use taxid limitations to pull bioB sequences from bvbrc query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "6a166184",
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_map = pd.read_csv(\"motif_prev/actinomycetota_bvbrc_reference_genomes.csv\")\n",
    "biob_patric = pd.read_csv(\"motif_prev/PGF_01400330_family_feature_list_full_seq.tsv\", delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "6136de73",
   "metadata": {},
   "outputs": [],
   "source": [
    "# genome_map.head()\n",
    "genome_map = genome_map[['Genome Name', 'NCBI Taxon ID', 'Genome ID', 'Order', 'Family', 'Genus']]\n",
    "\n",
    "biob_subset = biob_subset.merge(genome_map, how='left', left_on='staxid', right_on='NCBI Taxon ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "4be0c425",
   "metadata": {},
   "outputs": [],
   "source": [
    "bsap_phylogeny_genus_list = pd.Series(bsap_phylogeny_tax_list).str.split(\" \", expand=True)[0].unique()\n",
    "biob_subset = biob_subset[~biob_subset['Genus'].isin(bsap_phylogeny_genus_list)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "132c8919",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_subset = biob_subset.merge(biob_patric[['feature.genome_id', 'feature.patric_id', 'feature.na_sequence', 'feature.aa_sequence']],\n",
    "                 how='left', left_on='Genome ID', right_on='feature.genome_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "3599ef32",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_subset = biob_subset.dropna().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "f68aae6b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1275, 21)"
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "fc52fc4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>...</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>NCBI Taxon ID</th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>feature.genome_id</th>\n",
       "      <th>feature.patric_id</th>\n",
       "      <th>feature.na_sequence</th>\n",
       "      <th>feature.aa_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gi|1888951912|ref|NZ_JACGWZ010000001.1|</td>\n",
       "      <td>1.730000e-05</td>\n",
       "      <td>48.2</td>\n",
       "      <td>76.543</td>\n",
       "      <td>81</td>\n",
       "      <td>1000566</td>\n",
       "      <td>Halosaccharopolyspora lacisalsi</td>\n",
       "      <td>Halosaccharopolyspora lacisalsi</td>\n",
       "      <td>1356959</td>\n",
       "      <td>1357038</td>\n",
       "      <td>...</td>\n",
       "      <td>Saccharopolyspora lacisalsi strain DSM 45975</td>\n",
       "      <td>1000566</td>\n",
       "      <td>1000566.30</td>\n",
       "      <td>Pseudonocardiales</td>\n",
       "      <td>Pseudonocardiaceae</td>\n",
       "      <td>Halosaccharopolyspora</td>\n",
       "      <td>1000566.30</td>\n",
       "      <td>fig|1000566.3.peg.1406</td>\n",
       "      <td>atgagctccaccttccagcagctcgccgattcggtcctcgtcggga...</td>\n",
       "      <td>MSSTFQQLADSVLVGTPATHDDALAVLRADEAELMSLVAAAGRLRR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gi|1124504080|ref|NZ_FOJD01000002.1|</td>\n",
       "      <td>2.100000e-02</td>\n",
       "      <td>37.4</td>\n",
       "      <td>87.500</td>\n",
       "      <td>32</td>\n",
       "      <td>100225</td>\n",
       "      <td>Austwickia chelonae</td>\n",
       "      <td>Austwickia chelonae</td>\n",
       "      <td>539733</td>\n",
       "      <td>539763</td>\n",
       "      <td>...</td>\n",
       "      <td>Austwickia chelonae strain DSM 44178</td>\n",
       "      <td>100225</td>\n",
       "      <td>100225.50</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Dermatophilaceae</td>\n",
       "      <td>Austwickia</td>\n",
       "      <td>100225.50</td>\n",
       "      <td>fig|100225.5.peg.659</td>\n",
       "      <td>atgaccaccctcgatgtgaccgagacgaccggtacctccgagacgg...</td>\n",
       "      <td>MTTLDVTETTGTSETADLQHDGPTLDLTAMVEAGLAGRSITREQAM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gi|1441077360|ref|NZ_QQZY01000004.1|</td>\n",
       "      <td>1.550000e-09</td>\n",
       "      <td>60.8</td>\n",
       "      <td>69.484</td>\n",
       "      <td>213</td>\n",
       "      <td>1002870</td>\n",
       "      <td>Gaiella occulta</td>\n",
       "      <td>Gaiella occulta</td>\n",
       "      <td>267788</td>\n",
       "      <td>267995</td>\n",
       "      <td>...</td>\n",
       "      <td>Gaiella occulta strain F2-233</td>\n",
       "      <td>1002870</td>\n",
       "      <td>1002870.30</td>\n",
       "      <td>Gaiellales</td>\n",
       "      <td>Gaiellaceae</td>\n",
       "      <td>Gaiella</td>\n",
       "      <td>1002870.30</td>\n",
       "      <td>fig|1002870.3.peg.2128</td>\n",
       "      <td>atgcgtgacacacttcccccgatgcctgcagccgggtcgatccgcc...</td>\n",
       "      <td>MRDTLPPMPAAGSIRHDWTLREIEEIHALPLPELLFRAQSVHRAHH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gi|357407371|ref|NC_016113.1|</td>\n",
       "      <td>9.110000e-05</td>\n",
       "      <td>47.3</td>\n",
       "      <td>75.294</td>\n",
       "      <td>85</td>\n",
       "      <td>1003195</td>\n",
       "      <td>Streptantibioticus cattleyicolor NRRL 8057 = D...</td>\n",
       "      <td>Streptantibioticus cattleyicolor NRRL 8057 = D...</td>\n",
       "      <td>1386387</td>\n",
       "      <td>1386469</td>\n",
       "      <td>...</td>\n",
       "      <td>Streptomyces cattleya NRRL 8057 = DSM 46488</td>\n",
       "      <td>1003195</td>\n",
       "      <td>1003195.11</td>\n",
       "      <td>Streptomycetales</td>\n",
       "      <td>Streptomycetaceae</td>\n",
       "      <td>Streptantibioticus</td>\n",
       "      <td>1003195.11</td>\n",
       "      <td>fig|1003195.11.peg.1990</td>\n",
       "      <td>atggacctgctgaagacgctggtggacaagggactgcggggcgagt...</td>\n",
       "      <td>MDLLKTLVDKGLRGESPTREEALAVLATSDDELLDVVAAAGRVRRR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gi|357407371|ref|NC_016113.1|</td>\n",
       "      <td>9.110000e-05</td>\n",
       "      <td>47.3</td>\n",
       "      <td>75.294</td>\n",
       "      <td>85</td>\n",
       "      <td>1003195</td>\n",
       "      <td>Streptantibioticus cattleyicolor NRRL 8057 = D...</td>\n",
       "      <td>Streptantibioticus cattleyicolor NRRL 8057 = D...</td>\n",
       "      <td>1386387</td>\n",
       "      <td>1386469</td>\n",
       "      <td>...</td>\n",
       "      <td>Streptomyces cattleya NRRL 8057 = DSM 46488</td>\n",
       "      <td>1003195</td>\n",
       "      <td>1003195.11</td>\n",
       "      <td>Streptomycetales</td>\n",
       "      <td>Streptomycetaceae</td>\n",
       "      <td>Streptantibioticus</td>\n",
       "      <td>1003195.11</td>\n",
       "      <td>fig|1003195.11.peg.1258</td>\n",
       "      <td>atgcaactactcgacaccctcgtcggcaaggcgttacgccgcgaaa...</td>\n",
       "      <td>MQLLDTLVGKALRRETPTREEALAVLRTEDDDLLDVVAAAFRVRHH...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     seqid        evalue  bitscore  pident  \\\n",
       "0  gi|1888951912|ref|NZ_JACGWZ010000001.1|  1.730000e-05      48.2  76.543   \n",
       "1     gi|1124504080|ref|NZ_FOJD01000002.1|  2.100000e-02      37.4  87.500   \n",
       "2     gi|1441077360|ref|NZ_QQZY01000004.1|  1.550000e-09      60.8  69.484   \n",
       "3            gi|357407371|ref|NC_016113.1|  9.110000e-05      47.3  75.294   \n",
       "4            gi|357407371|ref|NC_016113.1|  9.110000e-05      47.3  75.294   \n",
       "\n",
       "   length   staxid                                             ssname  \\\n",
       "0      81  1000566                    Halosaccharopolyspora lacisalsi   \n",
       "1      32   100225                                Austwickia chelonae   \n",
       "2     213  1002870                                    Gaiella occulta   \n",
       "3      85  1003195  Streptantibioticus cattleyicolor NRRL 8057 = D...   \n",
       "4      85  1003195  Streptantibioticus cattleyicolor NRRL 8057 = D...   \n",
       "\n",
       "                                            scomname   sstart    ssend  ...  \\\n",
       "0                    Halosaccharopolyspora lacisalsi  1356959  1357038  ...   \n",
       "1                                Austwickia chelonae   539733   539763  ...   \n",
       "2                                    Gaiella occulta   267788   267995  ...   \n",
       "3  Streptantibioticus cattleyicolor NRRL 8057 = D...  1386387  1386469  ...   \n",
       "4  Streptantibioticus cattleyicolor NRRL 8057 = D...  1386387  1386469  ...   \n",
       "\n",
       "                                    Genome Name NCBI Taxon ID   Genome ID  \\\n",
       "0  Saccharopolyspora lacisalsi strain DSM 45975       1000566  1000566.30   \n",
       "1          Austwickia chelonae strain DSM 44178        100225   100225.50   \n",
       "2                 Gaiella occulta strain F2-233       1002870  1002870.30   \n",
       "3   Streptomyces cattleya NRRL 8057 = DSM 46488       1003195  1003195.11   \n",
       "4   Streptomyces cattleya NRRL 8057 = DSM 46488       1003195  1003195.11   \n",
       "\n",
       "               Order              Family                  Genus  \\\n",
       "0  Pseudonocardiales  Pseudonocardiaceae  Halosaccharopolyspora   \n",
       "1      Micrococcales    Dermatophilaceae             Austwickia   \n",
       "2         Gaiellales         Gaiellaceae                Gaiella   \n",
       "3   Streptomycetales   Streptomycetaceae     Streptantibioticus   \n",
       "4   Streptomycetales   Streptomycetaceae     Streptantibioticus   \n",
       "\n",
       "  feature.genome_id        feature.patric_id  \\\n",
       "0        1000566.30   fig|1000566.3.peg.1406   \n",
       "1         100225.50     fig|100225.5.peg.659   \n",
       "2        1002870.30   fig|1002870.3.peg.2128   \n",
       "3        1003195.11  fig|1003195.11.peg.1990   \n",
       "4        1003195.11  fig|1003195.11.peg.1258   \n",
       "\n",
       "                                 feature.na_sequence  \\\n",
       "0  atgagctccaccttccagcagctcgccgattcggtcctcgtcggga...   \n",
       "1  atgaccaccctcgatgtgaccgagacgaccggtacctccgagacgg...   \n",
       "2  atgcgtgacacacttcccccgatgcctgcagccgggtcgatccgcc...   \n",
       "3  atggacctgctgaagacgctggtggacaagggactgcggggcgagt...   \n",
       "4  atgcaactactcgacaccctcgtcggcaaggcgttacgccgcgaaa...   \n",
       "\n",
       "                                 feature.aa_sequence  \n",
       "0  MSSTFQQLADSVLVGTPATHDDALAVLRADEAELMSLVAAAGRLRR...  \n",
       "1  MTTLDVTETTGTSETADLQHDGPTLDLTAMVEAGLAGRSITREQAM...  \n",
       "2  MRDTLPPMPAAGSIRHDWTLREIEEIHALPLPELLFRAQSVHRAHH...  \n",
       "3  MDLLKTLVDKGLRGESPTREEALAVLATSDDELLDVVAAAGRVRRR...  \n",
       "4  MQLLDTLVGKALRRETPTREEALAVLRTEDDDLLDVVAAAFRVRHH...  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "68874472",
   "metadata": {},
   "outputs": [],
   "source": [
    "duplicate_genes_to_drop = [\n",
    "    \"fig|1123320.3.peg.4007\",\n",
    "    \"fig|1381558.3.peg.5909\",\n",
    "    \"fig|1530042.3.peg.5524\"\n",
    "    \"fig|1894969.3.peg.5036\",\n",
    "    \"fig|2017.4.peg.2522\",\n",
    "    \"fig|2339229.3.peg.6990\",\n",
    "    \"fig|2697569.3.peg.1071\",\n",
    "    \"fig|2697569.3.peg.1072\",\n",
    "    \"fig|2697569.3.peg.1073\",\n",
    "    \"fig|2705253.3.peg.6283\",\n",
    "    \"fig|2705254.3.peg.6198\",\n",
    "    \"fig|469383.5.peg.1574\",\n",
    "    \"fig|47989.4.peg.5630\",\n",
    "    \"fig|516124.4.peg.3448\",\n",
    "    \"fig|556532.3.peg.2151\",\n",
    "    \"fig|589240.3.peg.7642\",\n",
    "    \"fig|795644.3.peg.4543\",\n",
    "    \"fig|795644.3.peg.4542\",\n",
    "    \"fig|861266.6.peg.2194\",\n",
    "    \"fig|1530042.3.peg.5524\",\n",
    "    \"fig|1894969.3.peg.5036\"\n",
    "    ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "e94baad6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# biob_subset[biob_subset.duplicated(subset=['Genome Name'], keep=False)][\n",
    "#     ['seqid', 'bitscore', 'pident', 'length', 'sstart', 'ssend', 'Genome Name', 'feature.genome_id', 'feature.patric_id']].tail(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "f7c8062d",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_subset = biob_subset[~biob_subset['feature.patric_id'].isin(duplicate_genes_to_drop)].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "895ec01d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# biob_subset[biob_subset['Genome Name'].duplicated(keep=False)]\n",
    "biob_subset = biob_subset.drop_duplicates(subset=['Genome Name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "202a6720",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset.isna().sum().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "id": "34d69b9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for i in biob_subset['Order'].unique():\n",
    "#     print(i)\n",
    "# biob_subset['Order'].value_counts()\n",
    "\n",
    "\n",
    "biob_ind_orders = [\n",
    "    \"Streptomycetales\", \n",
    "    \"Micromonosporales\",\n",
    "    \"Micrococcales\",\n",
    "    \"Pseudonocardiales\",\n",
    "    \"Propionibacteriales\",\n",
    "    \"Streptosporangiales\", \n",
    "    \"Glycomycetales\", \n",
    "    \"Cryptosporangiales\", \n",
    "    \"Acidimicrobiales\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "14a2a098",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_subset = biob_subset[biob_subset['Order'].isin(biob_ind_orders)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "4ba991d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_subset = biob_subset[biob_subset['length'] > 100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "7713bc25",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count     311.000000\n",
       "mean     1003.533762\n",
       "std        34.677193\n",
       "min       477.000000\n",
       "25%       996.000000\n",
       "50%       996.000000\n",
       "75%      1017.000000\n",
       "max      1047.000000\n",
       "Name: feature.na_sequence, dtype: float64"
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset['feature.na_sequence'].apply(len).describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c511938",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73707599",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "30f03f7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_subset = biob_subset.drop_duplicates(subset=['Genome Name']).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "5338047c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ecoli_protein = \"MAHRPRWTLSQVTELFEKPLLDLLFEAQQVHRQHFDPRQVQVSTLLSIKTGACPEDCKYCPQSSRYKTGLEAERLMEVEQVLESARKAKAAGSTRFCMGAAWKNPHERDMPYLEQMVQGVKAMGLEACMTLGTLSESQAQRLANAGLDYYNHNLDTSPEFYGNIITTRTYQERLDTLEKVREAGIKVCSGGIVGLGETVKDRAGLLLQLANLPTPPESVPINMLVKVKGTPLADNDDVDAFDFIRTIAVARIMMPTSYVRLSAGREQMNEQTQAMCFMAGANSIFYGCKLLTTPNPEEDKDLQLFRKLGLNPQQTAVLAGDNEQQQRLEQALMTPDTDEYYNAAAL\"\n",
    "ecoli_gene_len = len(ecoli_protein)*3\n",
    "\n",
    "biob_subset = biob_subset[biob_subset['feature.na_sequence'].apply(len) < 1050]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "730d32aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "id": "26c2f9f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_cols_of_interest = ['Genome Name', 'NCBI Taxon ID', 'Genome ID', \n",
    "                           'seqid', 'Order', 'Family', 'Genus', 'feature.patric_id', \n",
    "                           'scomname', 'bitscore', 'pident', 'sstart', 'ssend', 'length',\n",
    "                           'feature.na_sequence', 'feature.aa_sequence']\n",
    "\n",
    "biob_subset[subset_cols_of_interest].to_csv(\"motif/biob_independent_seqs_filter2.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "f9e8a4ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "# write_df_to_fasta(biob_subset, 'feature.na_sequence', 'Genome Name', \"motif/putative2_biob_independent_seqs_nt.fasta\")\n",
    "# write_df_to_fasta(biob_subset, 'feature.aa_sequence', 'Genome Name', \"motif/putative2_biob_independent_seqs_filter2_aa.fasta\")\n",
    "write_df_to_fasta(biob_subset, 'feature.aa_sequence', 'Genome Name', \"motif/putative2_biob_independent_seqs_filter3_aa.fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a67423ba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "id": "14e9fed9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# biob_subset['Genus'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "0f8d1f52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>...</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>NCBI Taxon ID</th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>feature.genome_id</th>\n",
       "      <th>feature.patric_id</th>\n",
       "      <th>feature.na_sequence</th>\n",
       "      <th>feature.aa_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>gi|499073258|ref|NZ_HF570956.1|</td>\n",
       "      <td>1.160000e-169</td>\n",
       "      <td>592.0</td>\n",
       "      <td>73.887</td>\n",
       "      <td>1011</td>\n",
       "      <td>1193181</td>\n",
       "      <td>Phycicoccus elongatus Lp2</td>\n",
       "      <td>Phycicoccus elongatus Lp2</td>\n",
       "      <td>2909282</td>\n",
       "      <td>2910283</td>\n",
       "      <td>...</td>\n",
       "      <td>Tetrasphaera elongata Lp2</td>\n",
       "      <td>1193181</td>\n",
       "      <td>1193181.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Phycicoccus</td>\n",
       "      <td>1193181.30</td>\n",
       "      <td>fig|1193181.3.peg.715</td>\n",
       "      <td>atgaccacgacggcggaaaccacgaccgacatcctcgacgtcgccc...</td>\n",
       "      <td>MTTTAETTTDILDVARAQVLERGEALTQAQIVEVLRTGDDRLADLL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>gi|1910128291|ref|NZ_JACXBW010000017.1|</td>\n",
       "      <td>6.710000e-156</td>\n",
       "      <td>548.0</td>\n",
       "      <td>74.568</td>\n",
       "      <td>869</td>\n",
       "      <td>121292</td>\n",
       "      <td>Pseudarthrobacter sulfonivorans</td>\n",
       "      <td>Pseudarthrobacter sulfonivorans</td>\n",
       "      <td>1496</td>\n",
       "      <td>2357</td>\n",
       "      <td>...</td>\n",
       "      <td>Pseudarthrobacter sulfonivorans strain ALL</td>\n",
       "      <td>121292</td>\n",
       "      <td>121292.70</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>121292.70</td>\n",
       "      <td>fig|121292.7.peg.1477</td>\n",
       "      <td>atgacgatgcagccagacctccaggccaaccccaacaccatcctcg...</td>\n",
       "      <td>MTMQPDLQANPNTILETARARVLEQGMGLSESQLVEILRLPDDALP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>gi|1148759302|ref|NZ_KV908325.1|</td>\n",
       "      <td>9.870000e-172</td>\n",
       "      <td>600.0</td>\n",
       "      <td>76.319</td>\n",
       "      <td>853</td>\n",
       "      <td>1531955</td>\n",
       "      <td>Sinomonas mesophila</td>\n",
       "      <td>Sinomonas mesophila</td>\n",
       "      <td>52898</td>\n",
       "      <td>52050</td>\n",
       "      <td>...</td>\n",
       "      <td>Sinomonas mesophila strain MPKL 26</td>\n",
       "      <td>1531955</td>\n",
       "      <td>1531955.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Sinomonas</td>\n",
       "      <td>1531955.30</td>\n",
       "      <td>fig|1531955.3.peg.1813</td>\n",
       "      <td>atgaccagcggctacgccattctcgagacggcccggaagcaggtcc...</td>\n",
       "      <td>MTSGYAILETARKQVLHEGRGLREAQILEVLNLPDEAIPAALQLAH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>gi|1905952393|ref|NZ_BMKV01000004.1|</td>\n",
       "      <td>6.120000e-124</td>\n",
       "      <td>442.0</td>\n",
       "      <td>70.672</td>\n",
       "      <td>982</td>\n",
       "      <td>158897</td>\n",
       "      <td>Pseudarthrobacter scleromae</td>\n",
       "      <td>Pseudarthrobacter scleromae</td>\n",
       "      <td>515291</td>\n",
       "      <td>516262</td>\n",
       "      <td>...</td>\n",
       "      <td>Pseudarthrobacter scleromae strain CGMCC 1.3601</td>\n",
       "      <td>158897</td>\n",
       "      <td>158897.40</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>158897.40</td>\n",
       "      <td>fig|158897.4.peg.2682</td>\n",
       "      <td>atgacaaccatttcccgcaccacggaatatccaattctggagaccg...</td>\n",
       "      <td>MTTISRTTEYPILETARQQVLEDGIGLTEAQLVEVLRLPDPAVPAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>gi|914681903|ref|NZ_LAIR01000002.1|</td>\n",
       "      <td>6.940000e-168</td>\n",
       "      <td>587.0</td>\n",
       "      <td>73.851</td>\n",
       "      <td>979</td>\n",
       "      <td>1631356</td>\n",
       "      <td>Luteipulveratus halotolerans</td>\n",
       "      <td>Luteipulveratus halotolerans</td>\n",
       "      <td>3964322</td>\n",
       "      <td>3963350</td>\n",
       "      <td>...</td>\n",
       "      <td>Luteipulveratus halotolerans strain C296001</td>\n",
       "      <td>1631356</td>\n",
       "      <td>1631356.50</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Dermacoccaceae</td>\n",
       "      <td>Luteipulveratus</td>\n",
       "      <td>1631356.50</td>\n",
       "      <td>fig|1631356.5.peg.4010</td>\n",
       "      <td>gtgacctcgaccacgacgacgacgtccatcctcgaccgcgcccgcg...</td>\n",
       "      <td>MTSTTTTTSILDRAREQVLDRGEALDEQQILEVLQTSDDQLEPLLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>gi|1823028355|ref|NZ_JAAOIV010000013.1|</td>\n",
       "      <td>2.160000e-160</td>\n",
       "      <td>563.0</td>\n",
       "      <td>74.646</td>\n",
       "      <td>919</td>\n",
       "      <td>1656884</td>\n",
       "      <td>Metallococcus carri</td>\n",
       "      <td>Metallococcus carri</td>\n",
       "      <td>108669</td>\n",
       "      <td>107766</td>\n",
       "      <td>...</td>\n",
       "      <td>Calidifontibacter sp. DB0510</td>\n",
       "      <td>1656884</td>\n",
       "      <td>1656884.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Dermacoccaceae</td>\n",
       "      <td>Metallococcus</td>\n",
       "      <td>1656884.30</td>\n",
       "      <td>fig|1656884.3.peg.1085</td>\n",
       "      <td>atgagcgagaccaccgcgcccgcgtccgtgctcgatccggctcgtg...</td>\n",
       "      <td>MSETTAPASVLDPAREVLDRARETVLVRGEPLGYADLVEILRTPDE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>gi|1721967527|ref|NZ_VCQV01000004.1|</td>\n",
       "      <td>4.280000e-146</td>\n",
       "      <td>516.0</td>\n",
       "      <td>73.118</td>\n",
       "      <td>930</td>\n",
       "      <td>1660198</td>\n",
       "      <td>Leekyejoonella antrihumi</td>\n",
       "      <td>Leekyejoonella antrihumi</td>\n",
       "      <td>60703</td>\n",
       "      <td>59782</td>\n",
       "      <td>...</td>\n",
       "      <td>Dermacoccaceae bacterium C5-26</td>\n",
       "      <td>1660198</td>\n",
       "      <td>1660198.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Dermacoccaceae</td>\n",
       "      <td>Leekyejoonella</td>\n",
       "      <td>1660198.30</td>\n",
       "      <td>fig|1660198.3.peg.3462</td>\n",
       "      <td>atgacttcgaccaccaatgtcacctccatcctcgaccgcgcacgcg...</td>\n",
       "      <td>MTSTTNVTSILDRARDRVLVQGERLGYDDLVEVLQTGDDQLEDLLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>gi|1852738371|ref|NZ_JABTYH010000004.1|</td>\n",
       "      <td>3.310000e-134</td>\n",
       "      <td>476.0</td>\n",
       "      <td>73.309</td>\n",
       "      <td>828</td>\n",
       "      <td>1671</td>\n",
       "      <td>Pseudarthrobacter oxydans</td>\n",
       "      <td>Pseudarthrobacter oxydans</td>\n",
       "      <td>161419</td>\n",
       "      <td>162243</td>\n",
       "      <td>...</td>\n",
       "      <td>Pseudarthrobacter oxydans strain USM2</td>\n",
       "      <td>1671</td>\n",
       "      <td>1671.13</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>1671.13</td>\n",
       "      <td>fig|1671.13.peg.3183</td>\n",
       "      <td>atgacaaccatttcccgcaccacggaatacccaatcctggagacgg...</td>\n",
       "      <td>MTTISRTTEYPILETARQQVLEGGIGLAEAQLLEVLRLPDPAVPAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>gi|1906036206|ref|NZ_BMKU01000006.1|</td>\n",
       "      <td>7.230000e-117</td>\n",
       "      <td>417.0</td>\n",
       "      <td>71.347</td>\n",
       "      <td>876</td>\n",
       "      <td>1676</td>\n",
       "      <td>Pseudarthrobacter polychromogenes</td>\n",
       "      <td>Pseudarthrobacter polychromogenes</td>\n",
       "      <td>121353</td>\n",
       "      <td>120488</td>\n",
       "      <td>...</td>\n",
       "      <td>Pseudarthrobacter polychromogenes strain CGMCC...</td>\n",
       "      <td>1676</td>\n",
       "      <td>1676.60</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>1676.60</td>\n",
       "      <td>fig|1676.6.peg.2354</td>\n",
       "      <td>atgacaaccatttcccccaccacggaatatccaatcctggagaccg...</td>\n",
       "      <td>MTTISPTTEYPILETARQQVLEDGIGLTEAQLLEVLRLPDPAVPAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>gi|1317271524|ref|NZ_PJNE01000001.1|</td>\n",
       "      <td>7.070000e-173</td>\n",
       "      <td>603.0</td>\n",
       "      <td>74.419</td>\n",
       "      <td>989</td>\n",
       "      <td>173053</td>\n",
       "      <td>Phycicoccus duodecadis</td>\n",
       "      <td>Phycicoccus duodecadis</td>\n",
       "      <td>1986459</td>\n",
       "      <td>1985481</td>\n",
       "      <td>...</td>\n",
       "      <td>Tetrasphaera duodecadis strain DSM 12806</td>\n",
       "      <td>173053</td>\n",
       "      <td>173053.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Phycicoccus</td>\n",
       "      <td>173053.30</td>\n",
       "      <td>fig|173053.3.peg.1892</td>\n",
       "      <td>atgacggctcaggtgaccggaaccgacatcctcgacgtcgcgcgcg...</td>\n",
       "      <td>MTAQVTGTDILDVAREQVLERGVALTHEQIVAVLETGDDRLQDLLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>gi|1869599055|ref|NZ_JACCFW010000001.1|</td>\n",
       "      <td>5.420000e-162</td>\n",
       "      <td>568.0</td>\n",
       "      <td>74.480</td>\n",
       "      <td>913</td>\n",
       "      <td>1874116</td>\n",
       "      <td>Allobranchiibius huperziae</td>\n",
       "      <td>Allobranchiibius huperziae</td>\n",
       "      <td>122920</td>\n",
       "      <td>122017</td>\n",
       "      <td>...</td>\n",
       "      <td>Allobranchiibius huperziae strain DSM 29531</td>\n",
       "      <td>1874116</td>\n",
       "      <td>1874116.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Dermacoccaceae</td>\n",
       "      <td>Allobranchiibius</td>\n",
       "      <td>1874116.30</td>\n",
       "      <td>fig|1874116.3.peg.131</td>\n",
       "      <td>atgacctcgacaacgaccccccgcgacaccaccgtcctggaccgcg...</td>\n",
       "      <td>MTSTTTPRDTTVLDRAREVVLDRGEALGYDDIVAVLNTPDELLPDL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>gi|1821084693|ref|NZ_SAYU02000032.1|</td>\n",
       "      <td>5.080000e-163</td>\n",
       "      <td>572.0</td>\n",
       "      <td>74.207</td>\n",
       "      <td>977</td>\n",
       "      <td>2502783</td>\n",
       "      <td>Phycicoccus flavus</td>\n",
       "      <td>Phycicoccus flavus</td>\n",
       "      <td>1438</td>\n",
       "      <td>2402</td>\n",
       "      <td>...</td>\n",
       "      <td>Phycicoccus sp. CMS6Z-2</td>\n",
       "      <td>2502783</td>\n",
       "      <td>2502783.60</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Phycicoccus</td>\n",
       "      <td>2502783.60</td>\n",
       "      <td>fig|2502783.6.peg.1967</td>\n",
       "      <td>atgcacggcatgacggctcaggcgaccggcacggacatcctcgacg...</td>\n",
       "      <td>MHGMTAQATGTDILDVAREQVLGRGVPLTKEQVLDVLRTGDDRLQD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>gi|1801008088|ref|NZ_CP047898.1|</td>\n",
       "      <td>8.590000e-155</td>\n",
       "      <td>545.0</td>\n",
       "      <td>74.539</td>\n",
       "      <td>868</td>\n",
       "      <td>2697569</td>\n",
       "      <td>Pseudarthrobacter psychrotolerans</td>\n",
       "      <td>Pseudarthrobacter psychrotolerans</td>\n",
       "      <td>679410</td>\n",
       "      <td>680271</td>\n",
       "      <td>...</td>\n",
       "      <td>Pseudarthrobacter sp. YJ56</td>\n",
       "      <td>2697569</td>\n",
       "      <td>2697569.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>2697569.30</td>\n",
       "      <td>fig|2697569.3.peg.721</td>\n",
       "      <td>atgacgattcaggcacttcccacgaccaccaactacgcaattctcg...</td>\n",
       "      <td>MTIQALPTTTNYAILDAARRQVLDDGIGLSEDQLVEILQLPDEALP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>gi|2049880501|ref|NZ_JAHHWQ010000004.1|</td>\n",
       "      <td>3.000000e-153</td>\n",
       "      <td>538.0</td>\n",
       "      <td>72.921</td>\n",
       "      <td>986</td>\n",
       "      <td>2840470</td>\n",
       "      <td>Phycicoccus mangrovi</td>\n",
       "      <td>Phycicoccus mangrovi</td>\n",
       "      <td>55070</td>\n",
       "      <td>54092</td>\n",
       "      <td>...</td>\n",
       "      <td>Phycicoccus mangrovi MAQZ13P-2</td>\n",
       "      <td>2840470</td>\n",
       "      <td>2840470.40</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Phycicoccus</td>\n",
       "      <td>2840470.40</td>\n",
       "      <td>fig|2840470.4.peg.1895</td>\n",
       "      <td>atgaccgcacaggccaccggcaccgacatcctcgacgtcgcccgcg...</td>\n",
       "      <td>MTAQATGTDILDVAREQVLERGVPLTQEQVLEVLRTGDDRLQDLLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>gi|2189799284|ref|NZ_CP089293.1|</td>\n",
       "      <td>1.260000e-126</td>\n",
       "      <td>450.0</td>\n",
       "      <td>70.742</td>\n",
       "      <td>957</td>\n",
       "      <td>29320</td>\n",
       "      <td>Paenarthrobacter nicotinovorans</td>\n",
       "      <td>Paenarthrobacter nicotinovorans</td>\n",
       "      <td>1077010</td>\n",
       "      <td>1076056</td>\n",
       "      <td>...</td>\n",
       "      <td>Paenarthrobacter nicotinovorans ATCC 49919</td>\n",
       "      <td>29320</td>\n",
       "      <td>29320.60</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Paenarthrobacter</td>\n",
       "      <td>29320.60</td>\n",
       "      <td>fig|29320.60.peg.1024</td>\n",
       "      <td>atgacccacgcaccccaggaaaagaccgtccacgccattcttgaca...</td>\n",
       "      <td>MTHAPQEKTVHAILDTARRQVLKRGEGLNEDQLVEILRLPDDAIPA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>gi|1701234082|ref|NZ_VFOQ01000002.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>685.0</td>\n",
       "      <td>75.834</td>\n",
       "      <td>989</td>\n",
       "      <td>297536</td>\n",
       "      <td>Oryzihumus leptocrescens</td>\n",
       "      <td>Oryzihumus leptocrescens</td>\n",
       "      <td>105831</td>\n",
       "      <td>104846</td>\n",
       "      <td>...</td>\n",
       "      <td>Oryzihumus leptocrescens strain DSM 18082</td>\n",
       "      <td>297536</td>\n",
       "      <td>297536.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Oryzihumus</td>\n",
       "      <td>297536.30</td>\n",
       "      <td>fig|297536.3.peg.3570</td>\n",
       "      <td>atgactgccgttgagactgacattctggccgtcgcgcgcgagcagg...</td>\n",
       "      <td>MTAVETDILAVAREQVLERGEALRYEQILAVLNTPDEQLEELLALA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>gi|1695362700|ref|NZ_VHJD01000009.1|</td>\n",
       "      <td>8.440000e-129</td>\n",
       "      <td>457.0</td>\n",
       "      <td>72.479</td>\n",
       "      <td>843</td>\n",
       "      <td>361575</td>\n",
       "      <td>Pseudarthrobacter phenanthrenivorans</td>\n",
       "      <td>Pseudarthrobacter phenanthrenivorans</td>\n",
       "      <td>125266</td>\n",
       "      <td>124429</td>\n",
       "      <td>...</td>\n",
       "      <td>Pseudarthrobacter phenanthrenivorans strain MHSD1</td>\n",
       "      <td>361575</td>\n",
       "      <td>361575.16</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>361575.16</td>\n",
       "      <td>fig|361575.16.peg.4145</td>\n",
       "      <td>atgacaaatcatgcattccgggaatatcccattctccagaccgccc...</td>\n",
       "      <td>MTNHAFREYPILQTARQQVLEQGIGLTEAQLVEVLRLPDHAVPAAL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>gi|663751679|ref|NZ_JOEE01000002.1|</td>\n",
       "      <td>1.200000e-151</td>\n",
       "      <td>533.0</td>\n",
       "      <td>72.948</td>\n",
       "      <td>987</td>\n",
       "      <td>367299</td>\n",
       "      <td>Phycicoccus jejuensis</td>\n",
       "      <td>Phycicoccus jejuensis</td>\n",
       "      <td>226318</td>\n",
       "      <td>227296</td>\n",
       "      <td>...</td>\n",
       "      <td>Phycicoccus jejuensis strain NRRL B-24460</td>\n",
       "      <td>367299</td>\n",
       "      <td>367299.40</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Phycicoccus</td>\n",
       "      <td>367299.40</td>\n",
       "      <td>fig|367299.4.peg.913</td>\n",
       "      <td>atgacggcacaggccaccggcaccgacatcctcgacgtcgcccgcg...</td>\n",
       "      <td>MTAQATGTDILDVAREQVLERGVPLTQEQVLEVLRTGDERLQDLLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>gi|1958605779|ref|NZ_CP068013.1|</td>\n",
       "      <td>2.100000e-136</td>\n",
       "      <td>483.0</td>\n",
       "      <td>72.769</td>\n",
       "      <td>863</td>\n",
       "      <td>37931</td>\n",
       "      <td>Paenarthrobacter ureafaciens</td>\n",
       "      <td>Paenarthrobacter ureafaciens</td>\n",
       "      <td>1059918</td>\n",
       "      <td>1059058</td>\n",
       "      <td>...</td>\n",
       "      <td>Paenarthrobacter ureafaciens strain CZY1</td>\n",
       "      <td>37931</td>\n",
       "      <td>37931.15</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Paenarthrobacter</td>\n",
       "      <td>37931.15</td>\n",
       "      <td>fig|37931.15.peg.990</td>\n",
       "      <td>atgaccatcgacgcaacgcatgaagccactggttacgccattctgg...</td>\n",
       "      <td>MTIDATHEATGYAILDLAKDQVLERGEGLTEDQLVEILQLPDDAIP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>gi|1449840604|ref|NZ_QTUA01000001.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>650.0</td>\n",
       "      <td>75.230</td>\n",
       "      <td>977</td>\n",
       "      <td>419650</td>\n",
       "      <td>Calidifontibacter indicus</td>\n",
       "      <td>Calidifontibacter indicus</td>\n",
       "      <td>750880</td>\n",
       "      <td>749907</td>\n",
       "      <td>...</td>\n",
       "      <td>Calidifontibacter indicus strain DSM 22967</td>\n",
       "      <td>419650</td>\n",
       "      <td>419650.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Dermacoccaceae</td>\n",
       "      <td>Calidifontibacter</td>\n",
       "      <td>419650.30</td>\n",
       "      <td>fig|419650.3.peg.738</td>\n",
       "      <td>atgacgagcaccacatcgcagacctcgatcctcgaccgtgcccgtg...</td>\n",
       "      <td>MTSTTSQTSILDRARERVLVHGEPLAYDDLVAVLRTPDELLPDLLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>gi|1697809379|ref|NZ_BJMD01000022.1|</td>\n",
       "      <td>2.190000e-123</td>\n",
       "      <td>440.0</td>\n",
       "      <td>72.418</td>\n",
       "      <td>823</td>\n",
       "      <td>43663</td>\n",
       "      <td>Paenarthrobacter aurescens</td>\n",
       "      <td>Paenarthrobacter aurescens</td>\n",
       "      <td>47650</td>\n",
       "      <td>46831</td>\n",
       "      <td>...</td>\n",
       "      <td>Paenarthrobacter aurescens strain NBRC 12136</td>\n",
       "      <td>43663</td>\n",
       "      <td>43663.10</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Paenarthrobacter</td>\n",
       "      <td>43663.10</td>\n",
       "      <td>fig|43663.10.peg.3339</td>\n",
       "      <td>atgacggtccaagcagcccaaaaaccaggggaaacaactccccacg...</td>\n",
       "      <td>MTVQAAQKPGETTPHAILNIAREQVLERGQGLNEAQLKEVLELPDD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>gi|1906248993|ref|NZ_BMPM01000002.1|</td>\n",
       "      <td>2.070000e-130</td>\n",
       "      <td>462.0</td>\n",
       "      <td>71.281</td>\n",
       "      <td>968</td>\n",
       "      <td>43664</td>\n",
       "      <td>Paenarthrobacter histidinolovorans</td>\n",
       "      <td>Paenarthrobacter histidinolovorans</td>\n",
       "      <td>602178</td>\n",
       "      <td>603137</td>\n",
       "      <td>...</td>\n",
       "      <td>Paenarthrobacter histidinolovorans strain JCM ...</td>\n",
       "      <td>43664</td>\n",
       "      <td>43664.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Paenarthrobacter</td>\n",
       "      <td>43664.30</td>\n",
       "      <td>fig|43664.3.peg.1469</td>\n",
       "      <td>atgacccatgcaccccaggaagtgaccggctacgccattctcgaca...</td>\n",
       "      <td>MTHAPQEVTGYAILDTARRQVLERGEGLKESQLVEILRLPDEAIPA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>gi|1701377043|ref|NZ_VFMO01000001.1|</td>\n",
       "      <td>3.460000e-164</td>\n",
       "      <td>575.0</td>\n",
       "      <td>74.567</td>\n",
       "      <td>924</td>\n",
       "      <td>587872</td>\n",
       "      <td>Yimella lutea</td>\n",
       "      <td>Yimella lutea</td>\n",
       "      <td>610199</td>\n",
       "      <td>609283</td>\n",
       "      <td>...</td>\n",
       "      <td>Yimella lutea strain DSM 19828</td>\n",
       "      <td>587872</td>\n",
       "      <td>587872.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Dermacoccaceae</td>\n",
       "      <td>Yimella</td>\n",
       "      <td>587872.30</td>\n",
       "      <td>fig|587872.3.peg.608</td>\n",
       "      <td>atgaccagcaccacagcgaacacctcgatcctcgaacgcgcgcggg...</td>\n",
       "      <td>MTSTTANTSILERARDRVLIKGEGLTYDEIVEVLRTPDEMLPDLLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>gi|1798732407|ref|NZ_VMSB01000001.1|</td>\n",
       "      <td>5.040000e-156</td>\n",
       "      <td>548.0</td>\n",
       "      <td>73.104</td>\n",
       "      <td>989</td>\n",
       "      <td>622664</td>\n",
       "      <td>Ornithinibacter aureus</td>\n",
       "      <td>Ornithinibacter aureus</td>\n",
       "      <td>2573384</td>\n",
       "      <td>2572403</td>\n",
       "      <td>...</td>\n",
       "      <td>Ornithinibacter aureus strain DSM 23364</td>\n",
       "      <td>622664</td>\n",
       "      <td>622664.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Ornithinibacter</td>\n",
       "      <td>622664.30</td>\n",
       "      <td>fig|622664.3.peg.2525</td>\n",
       "      <td>atgaccgcacaggtgaccgggaccgacatcctcgacgtggcacgcg...</td>\n",
       "      <td>MTAQVTGTDILDVAREQVLERGVPLTQEQIEQVLRTGDDQLQALLA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>gi|1223689458|ref|NZ_LT629779.1|</td>\n",
       "      <td>7.920000e-123</td>\n",
       "      <td>437.0</td>\n",
       "      <td>70.582</td>\n",
       "      <td>945</td>\n",
       "      <td>728066</td>\n",
       "      <td>Pseudarthrobacter equi</td>\n",
       "      <td>Pseudarthrobacter equi</td>\n",
       "      <td>1404447</td>\n",
       "      <td>1403505</td>\n",
       "      <td>...</td>\n",
       "      <td>Pseudarthrobacter equi strain IMMIB L-1606</td>\n",
       "      <td>728066</td>\n",
       "      <td>728066.40</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>728066.40</td>\n",
       "      <td>fig|728066.4.peg.1315</td>\n",
       "      <td>atgactgccaccgtccacaactggcccatcctcgagacagcccgcg...</td>\n",
       "      <td>MTATVHNWPILETARAQVLERGTGLTEEQLVEVLTLPDEALPRALE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>gi|960377173|ref|NZ_KQ758607.1|</td>\n",
       "      <td>1.110000e-139</td>\n",
       "      <td>494.0</td>\n",
       "      <td>72.669</td>\n",
       "      <td>933</td>\n",
       "      <td>993070</td>\n",
       "      <td>Pseudarthrobacter enclensis</td>\n",
       "      <td>Pseudarthrobacter enclensis</td>\n",
       "      <td>109046</td>\n",
       "      <td>108122</td>\n",
       "      <td>...</td>\n",
       "      <td>Arthrobacter enclensis strain NIO-1008</td>\n",
       "      <td>993070</td>\n",
       "      <td>993070.40</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Pseudarthrobacter</td>\n",
       "      <td>993070.40</td>\n",
       "      <td>fig|993070.4.peg.3637</td>\n",
       "      <td>atgaccatccacgctacagaccacgccatcctcgacaccgcgcgcc...</td>\n",
       "      <td>MTIHATDHAILDTARRQVLDQGIGLAEEQLVDVLSLPDDALPAALD...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       seqid         evalue  bitscore  pident  \\\n",
       "40           gi|499073258|ref|NZ_HF570956.1|  1.160000e-169     592.0  73.887   \n",
       "42   gi|1910128291|ref|NZ_JACXBW010000017.1|  6.710000e-156     548.0  74.568   \n",
       "96          gi|1148759302|ref|NZ_KV908325.1|  9.870000e-172     600.0  76.319   \n",
       "100     gi|1905952393|ref|NZ_BMKV01000004.1|  6.120000e-124     442.0  70.672   \n",
       "106      gi|914681903|ref|NZ_LAIR01000002.1|  6.940000e-168     587.0  73.851   \n",
       "109  gi|1823028355|ref|NZ_JAAOIV010000013.1|  2.160000e-160     563.0  74.646   \n",
       "110     gi|1721967527|ref|NZ_VCQV01000004.1|  4.280000e-146     516.0  73.118   \n",
       "111  gi|1852738371|ref|NZ_JABTYH010000004.1|  3.310000e-134     476.0  73.309   \n",
       "112     gi|1906036206|ref|NZ_BMKU01000006.1|  7.230000e-117     417.0  71.347   \n",
       "114     gi|1317271524|ref|NZ_PJNE01000001.1|  7.070000e-173     603.0  74.419   \n",
       "127  gi|1869599055|ref|NZ_JACCFW010000001.1|  5.420000e-162     568.0  74.480   \n",
       "169     gi|1821084693|ref|NZ_SAYU02000032.1|  5.080000e-163     572.0  74.207   \n",
       "181         gi|1801008088|ref|NZ_CP047898.1|  8.590000e-155     545.0  74.539   \n",
       "197  gi|2049880501|ref|NZ_JAHHWQ010000004.1|  3.000000e-153     538.0  72.921   \n",
       "205         gi|2189799284|ref|NZ_CP089293.1|  1.260000e-126     450.0  70.742   \n",
       "206     gi|1701234082|ref|NZ_VFOQ01000002.1|   0.000000e+00     685.0  75.834   \n",
       "228     gi|1695362700|ref|NZ_VHJD01000009.1|  8.440000e-129     457.0  72.479   \n",
       "229      gi|663751679|ref|NZ_JOEE01000002.1|  1.200000e-151     533.0  72.948   \n",
       "233         gi|1958605779|ref|NZ_CP068013.1|  2.100000e-136     483.0  72.769   \n",
       "236     gi|1449840604|ref|NZ_QTUA01000001.1|   0.000000e+00     650.0  75.230   \n",
       "242     gi|1697809379|ref|NZ_BJMD01000022.1|  2.190000e-123     440.0  72.418   \n",
       "243     gi|1906248993|ref|NZ_BMPM01000002.1|  2.070000e-130     462.0  71.281   \n",
       "294     gi|1701377043|ref|NZ_VFMO01000001.1|  3.460000e-164     575.0  74.567   \n",
       "301     gi|1798732407|ref|NZ_VMSB01000001.1|  5.040000e-156     548.0  73.104   \n",
       "332         gi|1223689458|ref|NZ_LT629779.1|  7.920000e-123     437.0  70.582   \n",
       "357          gi|960377173|ref|NZ_KQ758607.1|  1.110000e-139     494.0  72.669   \n",
       "\n",
       "     length   staxid                                ssname  \\\n",
       "40     1011  1193181             Phycicoccus elongatus Lp2   \n",
       "42      869   121292       Pseudarthrobacter sulfonivorans   \n",
       "96      853  1531955                   Sinomonas mesophila   \n",
       "100     982   158897           Pseudarthrobacter scleromae   \n",
       "106     979  1631356          Luteipulveratus halotolerans   \n",
       "109     919  1656884                   Metallococcus carri   \n",
       "110     930  1660198              Leekyejoonella antrihumi   \n",
       "111     828     1671             Pseudarthrobacter oxydans   \n",
       "112     876     1676     Pseudarthrobacter polychromogenes   \n",
       "114     989   173053                Phycicoccus duodecadis   \n",
       "127     913  1874116            Allobranchiibius huperziae   \n",
       "169     977  2502783                    Phycicoccus flavus   \n",
       "181     868  2697569     Pseudarthrobacter psychrotolerans   \n",
       "197     986  2840470                  Phycicoccus mangrovi   \n",
       "205     957    29320       Paenarthrobacter nicotinovorans   \n",
       "206     989   297536              Oryzihumus leptocrescens   \n",
       "228     843   361575  Pseudarthrobacter phenanthrenivorans   \n",
       "229     987   367299                 Phycicoccus jejuensis   \n",
       "233     863    37931          Paenarthrobacter ureafaciens   \n",
       "236     977   419650             Calidifontibacter indicus   \n",
       "242     823    43663            Paenarthrobacter aurescens   \n",
       "243     968    43664    Paenarthrobacter histidinolovorans   \n",
       "294     924   587872                         Yimella lutea   \n",
       "301     989   622664                Ornithinibacter aureus   \n",
       "332     945   728066                Pseudarthrobacter equi   \n",
       "357     933   993070           Pseudarthrobacter enclensis   \n",
       "\n",
       "                                 scomname   sstart    ssend  ...  \\\n",
       "40              Phycicoccus elongatus Lp2  2909282  2910283  ...   \n",
       "42        Pseudarthrobacter sulfonivorans     1496     2357  ...   \n",
       "96                    Sinomonas mesophila    52898    52050  ...   \n",
       "100           Pseudarthrobacter scleromae   515291   516262  ...   \n",
       "106          Luteipulveratus halotolerans  3964322  3963350  ...   \n",
       "109                   Metallococcus carri   108669   107766  ...   \n",
       "110              Leekyejoonella antrihumi    60703    59782  ...   \n",
       "111             Pseudarthrobacter oxydans   161419   162243  ...   \n",
       "112     Pseudarthrobacter polychromogenes   121353   120488  ...   \n",
       "114                Phycicoccus duodecadis  1986459  1985481  ...   \n",
       "127            Allobranchiibius huperziae   122920   122017  ...   \n",
       "169                    Phycicoccus flavus     1438     2402  ...   \n",
       "181     Pseudarthrobacter psychrotolerans   679410   680271  ...   \n",
       "197                  Phycicoccus mangrovi    55070    54092  ...   \n",
       "205       Paenarthrobacter nicotinovorans  1077010  1076056  ...   \n",
       "206              Oryzihumus leptocrescens   105831   104846  ...   \n",
       "228  Pseudarthrobacter phenanthrenivorans   125266   124429  ...   \n",
       "229                 Phycicoccus jejuensis   226318   227296  ...   \n",
       "233          Paenarthrobacter ureafaciens  1059918  1059058  ...   \n",
       "236             Calidifontibacter indicus   750880   749907  ...   \n",
       "242            Paenarthrobacter aurescens    47650    46831  ...   \n",
       "243    Paenarthrobacter histidinolovorans   602178   603137  ...   \n",
       "294                         Yimella lutea   610199   609283  ...   \n",
       "301                Ornithinibacter aureus  2573384  2572403  ...   \n",
       "332                Pseudarthrobacter equi  1404447  1403505  ...   \n",
       "357           Pseudarthrobacter enclensis   109046   108122  ...   \n",
       "\n",
       "                                           Genome Name NCBI Taxon ID  \\\n",
       "40                           Tetrasphaera elongata Lp2       1193181   \n",
       "42          Pseudarthrobacter sulfonivorans strain ALL        121292   \n",
       "96                  Sinomonas mesophila strain MPKL 26       1531955   \n",
       "100    Pseudarthrobacter scleromae strain CGMCC 1.3601        158897   \n",
       "106        Luteipulveratus halotolerans strain C296001       1631356   \n",
       "109                       Calidifontibacter sp. DB0510       1656884   \n",
       "110                     Dermacoccaceae bacterium C5-26       1660198   \n",
       "111              Pseudarthrobacter oxydans strain USM2          1671   \n",
       "112  Pseudarthrobacter polychromogenes strain CGMCC...          1676   \n",
       "114           Tetrasphaera duodecadis strain DSM 12806        173053   \n",
       "127        Allobranchiibius huperziae strain DSM 29531       1874116   \n",
       "169                            Phycicoccus sp. CMS6Z-2       2502783   \n",
       "181                         Pseudarthrobacter sp. YJ56       2697569   \n",
       "197                     Phycicoccus mangrovi MAQZ13P-2       2840470   \n",
       "205         Paenarthrobacter nicotinovorans ATCC 49919         29320   \n",
       "206          Oryzihumus leptocrescens strain DSM 18082        297536   \n",
       "228  Pseudarthrobacter phenanthrenivorans strain MHSD1        361575   \n",
       "229          Phycicoccus jejuensis strain NRRL B-24460        367299   \n",
       "233           Paenarthrobacter ureafaciens strain CZY1         37931   \n",
       "236         Calidifontibacter indicus strain DSM 22967        419650   \n",
       "242       Paenarthrobacter aurescens strain NBRC 12136         43663   \n",
       "243  Paenarthrobacter histidinolovorans strain JCM ...         43664   \n",
       "294                     Yimella lutea strain DSM 19828        587872   \n",
       "301            Ornithinibacter aureus strain DSM 23364        622664   \n",
       "332         Pseudarthrobacter equi strain IMMIB L-1606        728066   \n",
       "357             Arthrobacter enclensis strain NIO-1008        993070   \n",
       "\n",
       "      Genome ID          Order              Family              Genus  \\\n",
       "40   1193181.30  Micrococcales  Intrasporangiaceae        Phycicoccus   \n",
       "42    121292.70  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "96   1531955.30  Micrococcales      Micrococcaceae          Sinomonas   \n",
       "100   158897.40  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "106  1631356.50  Micrococcales      Dermacoccaceae    Luteipulveratus   \n",
       "109  1656884.30  Micrococcales      Dermacoccaceae      Metallococcus   \n",
       "110  1660198.30  Micrococcales      Dermacoccaceae     Leekyejoonella   \n",
       "111     1671.13  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "112     1676.60  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "114   173053.30  Micrococcales  Intrasporangiaceae        Phycicoccus   \n",
       "127  1874116.30  Micrococcales      Dermacoccaceae   Allobranchiibius   \n",
       "169  2502783.60  Micrococcales  Intrasporangiaceae        Phycicoccus   \n",
       "181  2697569.30  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "197  2840470.40  Micrococcales  Intrasporangiaceae        Phycicoccus   \n",
       "205    29320.60  Micrococcales      Micrococcaceae   Paenarthrobacter   \n",
       "206   297536.30  Micrococcales  Intrasporangiaceae         Oryzihumus   \n",
       "228   361575.16  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "229   367299.40  Micrococcales  Intrasporangiaceae        Phycicoccus   \n",
       "233    37931.15  Micrococcales      Micrococcaceae   Paenarthrobacter   \n",
       "236   419650.30  Micrococcales      Dermacoccaceae  Calidifontibacter   \n",
       "242    43663.10  Micrococcales      Micrococcaceae   Paenarthrobacter   \n",
       "243    43664.30  Micrococcales      Micrococcaceae   Paenarthrobacter   \n",
       "294   587872.30  Micrococcales      Dermacoccaceae            Yimella   \n",
       "301   622664.30  Micrococcales  Intrasporangiaceae    Ornithinibacter   \n",
       "332   728066.40  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "357   993070.40  Micrococcales      Micrococcaceae  Pseudarthrobacter   \n",
       "\n",
       "    feature.genome_id       feature.patric_id  \\\n",
       "40         1193181.30   fig|1193181.3.peg.715   \n",
       "42          121292.70   fig|121292.7.peg.1477   \n",
       "96         1531955.30  fig|1531955.3.peg.1813   \n",
       "100         158897.40   fig|158897.4.peg.2682   \n",
       "106        1631356.50  fig|1631356.5.peg.4010   \n",
       "109        1656884.30  fig|1656884.3.peg.1085   \n",
       "110        1660198.30  fig|1660198.3.peg.3462   \n",
       "111           1671.13    fig|1671.13.peg.3183   \n",
       "112           1676.60     fig|1676.6.peg.2354   \n",
       "114         173053.30   fig|173053.3.peg.1892   \n",
       "127        1874116.30   fig|1874116.3.peg.131   \n",
       "169        2502783.60  fig|2502783.6.peg.1967   \n",
       "181        2697569.30   fig|2697569.3.peg.721   \n",
       "197        2840470.40  fig|2840470.4.peg.1895   \n",
       "205          29320.60   fig|29320.60.peg.1024   \n",
       "206         297536.30   fig|297536.3.peg.3570   \n",
       "228         361575.16  fig|361575.16.peg.4145   \n",
       "229         367299.40    fig|367299.4.peg.913   \n",
       "233          37931.15    fig|37931.15.peg.990   \n",
       "236         419650.30    fig|419650.3.peg.738   \n",
       "242          43663.10   fig|43663.10.peg.3339   \n",
       "243          43664.30    fig|43664.3.peg.1469   \n",
       "294         587872.30    fig|587872.3.peg.608   \n",
       "301         622664.30   fig|622664.3.peg.2525   \n",
       "332         728066.40   fig|728066.4.peg.1315   \n",
       "357         993070.40   fig|993070.4.peg.3637   \n",
       "\n",
       "                                   feature.na_sequence  \\\n",
       "40   atgaccacgacggcggaaaccacgaccgacatcctcgacgtcgccc...   \n",
       "42   atgacgatgcagccagacctccaggccaaccccaacaccatcctcg...   \n",
       "96   atgaccagcggctacgccattctcgagacggcccggaagcaggtcc...   \n",
       "100  atgacaaccatttcccgcaccacggaatatccaattctggagaccg...   \n",
       "106  gtgacctcgaccacgacgacgacgtccatcctcgaccgcgcccgcg...   \n",
       "109  atgagcgagaccaccgcgcccgcgtccgtgctcgatccggctcgtg...   \n",
       "110  atgacttcgaccaccaatgtcacctccatcctcgaccgcgcacgcg...   \n",
       "111  atgacaaccatttcccgcaccacggaatacccaatcctggagacgg...   \n",
       "112  atgacaaccatttcccccaccacggaatatccaatcctggagaccg...   \n",
       "114  atgacggctcaggtgaccggaaccgacatcctcgacgtcgcgcgcg...   \n",
       "127  atgacctcgacaacgaccccccgcgacaccaccgtcctggaccgcg...   \n",
       "169  atgcacggcatgacggctcaggcgaccggcacggacatcctcgacg...   \n",
       "181  atgacgattcaggcacttcccacgaccaccaactacgcaattctcg...   \n",
       "197  atgaccgcacaggccaccggcaccgacatcctcgacgtcgcccgcg...   \n",
       "205  atgacccacgcaccccaggaaaagaccgtccacgccattcttgaca...   \n",
       "206  atgactgccgttgagactgacattctggccgtcgcgcgcgagcagg...   \n",
       "228  atgacaaatcatgcattccgggaatatcccattctccagaccgccc...   \n",
       "229  atgacggcacaggccaccggcaccgacatcctcgacgtcgcccgcg...   \n",
       "233  atgaccatcgacgcaacgcatgaagccactggttacgccattctgg...   \n",
       "236  atgacgagcaccacatcgcagacctcgatcctcgaccgtgcccgtg...   \n",
       "242  atgacggtccaagcagcccaaaaaccaggggaaacaactccccacg...   \n",
       "243  atgacccatgcaccccaggaagtgaccggctacgccattctcgaca...   \n",
       "294  atgaccagcaccacagcgaacacctcgatcctcgaacgcgcgcggg...   \n",
       "301  atgaccgcacaggtgaccgggaccgacatcctcgacgtggcacgcg...   \n",
       "332  atgactgccaccgtccacaactggcccatcctcgagacagcccgcg...   \n",
       "357  atgaccatccacgctacagaccacgccatcctcgacaccgcgcgcc...   \n",
       "\n",
       "                                   feature.aa_sequence  \n",
       "40   MTTTAETTTDILDVARAQVLERGEALTQAQIVEVLRTGDDRLADLL...  \n",
       "42   MTMQPDLQANPNTILETARARVLEQGMGLSESQLVEILRLPDDALP...  \n",
       "96   MTSGYAILETARKQVLHEGRGLREAQILEVLNLPDEAIPAALQLAH...  \n",
       "100  MTTISRTTEYPILETARQQVLEDGIGLTEAQLVEVLRLPDPAVPAA...  \n",
       "106  MTSTTTTTSILDRAREQVLDRGEALDEQQILEVLQTSDDQLEPLLA...  \n",
       "109  MSETTAPASVLDPAREVLDRARETVLVRGEPLGYADLVEILRTPDE...  \n",
       "110  MTSTTNVTSILDRARDRVLVQGERLGYDDLVEVLQTGDDQLEDLLA...  \n",
       "111  MTTISRTTEYPILETARQQVLEGGIGLAEAQLLEVLRLPDPAVPAA...  \n",
       "112  MTTISPTTEYPILETARQQVLEDGIGLTEAQLLEVLRLPDPAVPAA...  \n",
       "114  MTAQVTGTDILDVAREQVLERGVALTHEQIVAVLETGDDRLQDLLA...  \n",
       "127  MTSTTTPRDTTVLDRAREVVLDRGEALGYDDIVAVLNTPDELLPDL...  \n",
       "169  MHGMTAQATGTDILDVAREQVLGRGVPLTKEQVLDVLRTGDDRLQD...  \n",
       "181  MTIQALPTTTNYAILDAARRQVLDDGIGLSEDQLVEILQLPDEALP...  \n",
       "197  MTAQATGTDILDVAREQVLERGVPLTQEQVLEVLRTGDDRLQDLLA...  \n",
       "205  MTHAPQEKTVHAILDTARRQVLKRGEGLNEDQLVEILRLPDDAIPA...  \n",
       "206  MTAVETDILAVAREQVLERGEALRYEQILAVLNTPDEQLEELLALA...  \n",
       "228  MTNHAFREYPILQTARQQVLEQGIGLTEAQLVEVLRLPDHAVPAAL...  \n",
       "229  MTAQATGTDILDVAREQVLERGVPLTQEQVLEVLRTGDERLQDLLA...  \n",
       "233  MTIDATHEATGYAILDLAKDQVLERGEGLTEDQLVEILQLPDDAIP...  \n",
       "236  MTSTTSQTSILDRARERVLVHGEPLAYDDLVAVLRTPDELLPDLLA...  \n",
       "242  MTVQAAQKPGETTPHAILNIAREQVLERGQGLNEAQLKEVLELPDD...  \n",
       "243  MTHAPQEVTGYAILDTARRQVLERGEGLKESQLVEILRLPDEAIPA...  \n",
       "294  MTSTTANTSILERARDRVLIKGEGLTYDEIVEVLRTPDEMLPDLLA...  \n",
       "301  MTAQVTGTDILDVAREQVLERGVPLTQEQIEQVLRTGDDQLQALLA...  \n",
       "332  MTATVHNWPILETARAQVLERGTGLTEEQLVEVLTLPDEALPRALE...  \n",
       "357  MTIHATDHAILDTARRQVLDQGIGLAEEQLVDVLSLPDDALPAALD...  \n",
       "\n",
       "[26 rows x 21 columns]"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset[biob_subset['Order'] == 'Micrococcales']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "534fc74e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(344, 21)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "95f25e79",
   "metadata": {},
   "outputs": [],
   "source": [
    "# biob_subset[biob_subset['Family'].str.contains('Pseudonocardiaceae')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "9843acb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# biob_dep[biob_dep['Family'].str.contains('Pseudonocardiaceae')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48110500",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e7494e1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73586266",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a22f460",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "035d6696",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>...</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>NCBI Taxon ID</th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>feature.genome_id</th>\n",
       "      <th>feature.patric_id</th>\n",
       "      <th>feature.na_sequence</th>\n",
       "      <th>feature.aa_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>gi|2179245779|ref|NZ_JAKEEC010000044.1|</td>\n",
       "      <td>1.610000e-137</td>\n",
       "      <td>487.0</td>\n",
       "      <td>73.241</td>\n",
       "      <td>867</td>\n",
       "      <td>369936</td>\n",
       "      <td>Arthrobacter alkaliphilus</td>\n",
       "      <td>Arthrobacter alkaliphilus</td>\n",
       "      <td>2145</td>\n",
       "      <td>3001</td>\n",
       "      <td>...</td>\n",
       "      <td>Arthrobacter alkaliphilus ZX_2022a</td>\n",
       "      <td>369936</td>\n",
       "      <td>369936.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Arthrobacter</td>\n",
       "      <td>369936.30</td>\n",
       "      <td>fig|369936.3.peg.3263</td>\n",
       "      <td>atgacgattcaggcagaccctcctgttaccagctacgccatcctcg...</td>\n",
       "      <td>MTIQADPPVTSYAILDTARKQVLEDGVGLTESQLVDVLRLPDDAVP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>gi|1869538579|ref|NZ_JACCAB010000001.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>667.0</td>\n",
       "      <td>75.767</td>\n",
       "      <td>978</td>\n",
       "      <td>642776</td>\n",
       "      <td>Pedococcus badiiscoriae</td>\n",
       "      <td>Pedococcus badiiscoriae</td>\n",
       "      <td>2487278</td>\n",
       "      <td>2488251</td>\n",
       "      <td>...</td>\n",
       "      <td>Phycicoccus badiiscoriae strain DSM 23987</td>\n",
       "      <td>642776</td>\n",
       "      <td>642776.30</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>Pedococcus</td>\n",
       "      <td>642776.30</td>\n",
       "      <td>fig|642776.3.peg.2394</td>\n",
       "      <td>atgacaattgaggagtcggttgccagcgacgtcctgcaggtcgctc...</td>\n",
       "      <td>MTIEESVASDVLQVAREQVLDRGEPLSHGQILDVLRTPDDQLEELL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>gi|1595894816|ref|NZ_SMZQ01000012.1|</td>\n",
       "      <td>2.340000e-161</td>\n",
       "      <td>565.0</td>\n",
       "      <td>73.684</td>\n",
       "      <td>969</td>\n",
       "      <td>683150</td>\n",
       "      <td>Arthrobacter nitrophenolicus</td>\n",
       "      <td>Arthrobacter nitrophenolicus</td>\n",
       "      <td>28527</td>\n",
       "      <td>27568</td>\n",
       "      <td>...</td>\n",
       "      <td>Arthrobacter nitrophenolicus strain S-A1</td>\n",
       "      <td>683150</td>\n",
       "      <td>683150.17</td>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>Micrococcaceae</td>\n",
       "      <td>Arthrobacter</td>\n",
       "      <td>683150.17</td>\n",
       "      <td>fig|683150.17.peg.367</td>\n",
       "      <td>atgacgattcaaccgcaactcctgaagcatccaatccttgagaccg...</td>\n",
       "      <td>MTIQPQLLKHPILETAREQVLEQGRGLSEAQLVEVLRLPDEALPAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       seqid         evalue  bitscore  pident  \\\n",
       "349  gi|2179245779|ref|NZ_JAKEEC010000044.1|  1.610000e-137     487.0  73.241   \n",
       "424  gi|1869538579|ref|NZ_JACCAB010000001.1|   0.000000e+00     667.0  75.767   \n",
       "430     gi|1595894816|ref|NZ_SMZQ01000012.1|  2.340000e-161     565.0  73.684   \n",
       "\n",
       "     length  staxid                        ssname  \\\n",
       "349     867  369936     Arthrobacter alkaliphilus   \n",
       "424     978  642776       Pedococcus badiiscoriae   \n",
       "430     969  683150  Arthrobacter nitrophenolicus   \n",
       "\n",
       "                         scomname   sstart    ssend  ...  \\\n",
       "349     Arthrobacter alkaliphilus     2145     3001  ...   \n",
       "424       Pedococcus badiiscoriae  2487278  2488251  ...   \n",
       "430  Arthrobacter nitrophenolicus    28527    27568  ...   \n",
       "\n",
       "                                   Genome Name NCBI Taxon ID  Genome ID  \\\n",
       "349         Arthrobacter alkaliphilus ZX_2022a        369936  369936.30   \n",
       "424  Phycicoccus badiiscoriae strain DSM 23987        642776  642776.30   \n",
       "430   Arthrobacter nitrophenolicus strain S-A1        683150  683150.17   \n",
       "\n",
       "             Order              Family         Genus feature.genome_id  \\\n",
       "349  Micrococcales      Micrococcaceae  Arthrobacter         369936.30   \n",
       "424  Micrococcales  Intrasporangiaceae    Pedococcus         642776.30   \n",
       "430  Micrococcales      Micrococcaceae  Arthrobacter         683150.17   \n",
       "\n",
       "         feature.patric_id                                feature.na_sequence  \\\n",
       "349  fig|369936.3.peg.3263  atgacgattcaggcagaccctcctgttaccagctacgccatcctcg...   \n",
       "424  fig|642776.3.peg.2394  atgacaattgaggagtcggttgccagcgacgtcctgcaggtcgctc...   \n",
       "430  fig|683150.17.peg.367  atgacgattcaaccgcaactcctgaagcatccaatccttgagaccg...   \n",
       "\n",
       "                                   feature.aa_sequence  \n",
       "349  MTIQADPPVTSYAILDTARKQVLEDGVGLTESQLVDVLRLPDDAVP...  \n",
       "424  MTIEESVASDVLQVAREQVLDRGEPLSHGQILDVLRTPDDQLEELL...  \n",
       "430  MTIQPQLLKHPILETAREQVLEQGRGLSEAQLVEVLRLPDEALPAA...  \n",
       "\n",
       "[3 rows x 21 columns]"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# biob_subset[subset_cols_of_interest]\n",
    "\n",
    "# biob_subset[biob_subset['Genus'].str.contains('Pedo')]\n",
    "# biob_subset[biob_subset['Family'] == 'Mycobacteriaceae']\n",
    "\n",
    "biob_dep[biob_dep['Order'] == 'Micrococcales']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "350f68bf",
   "metadata": {},
   "source": [
    "## generate biob bsap dependent dataset:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "f7fbd07a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## generate a proximity filter between biob and bsap positions:\n",
    "bsap = bsap.merge(biob[biob['length'] > 100][['staxid', 'pident', 'length', 'sstart', 'ssend']], on='staxid', suffixes=(\"_bsap\", \"_biob\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "1ad50e4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_feature_proximity(row):\n",
    "    bsap_start = min(row.sstart_bsap, row.ssend_bsap)\n",
    "    bsap_end = max(row.sstart_bsap, row.ssend_bsap)\n",
    "    \n",
    "    biob_start = min(row.sstart_biob, row.ssend_biob)\n",
    "    biob_end = max(row.sstart_biob, row.ssend_biob)\n",
    "    \n",
    "    \n",
    "    if biob_start < bsap_start:\n",
    "        prox = biob_start - biob_end\n",
    "        \n",
    "    elif bsap_start <= biob_start:\n",
    "        prox = biob_start - bsap_end\n",
    "        \n",
    "    return prox"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "id": "162f5dd1",
   "metadata": {},
   "outputs": [],
   "source": [
    "bsap['proximity'] = bsap.apply(get_feature_proximity, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "9f3a4f88",
   "metadata": {},
   "outputs": [],
   "source": [
    "bsap = bsap[bsap['proximity'].abs() < 1000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8af9aa26",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "81bba2e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_dep = biob[biob['staxid'].isin(bsap['staxid'].unique())].reset_index(drop=True).copy()\n",
    "\n",
    "# genome_map.head()\n",
    "genome_map = genome_map[['Genome Name', 'NCBI Taxon ID', 'Genome ID', 'Order', 'Family', 'Genus']]\n",
    "\n",
    "biob_dep = biob_dep.merge(genome_map, how='left', left_on='staxid', right_on='NCBI Taxon ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "9b0f8eda",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_independent_genus_list = biob_subset['Genus'].unique()\n",
    "\n",
    "biob_dep = biob_dep[~biob_dep['Genus'].isin(biob_independent_genus_list)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "01402f9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_dep = biob_dep.merge(biob_patric[['feature.genome_id', 'feature.patric_id', 'feature.na_sequence', 'feature.aa_sequence']],\n",
    "                 how='left', left_on='Genome ID', right_on='feature.genome_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "39f3c56f",
   "metadata": {},
   "outputs": [],
   "source": [
    "bsap_filter_df = bsap.merge(genome_map, how='left', left_on='staxid', right_on='NCBI Taxon ID').merge(\n",
    "    biob_patric[['feature.genome_id', 'feature.patric_id', 'feature.na_sequence', 'feature.aa_sequence']],\n",
    "                 how='left', left_on='Genome ID', right_on='feature.genome_id').copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "d1edca8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "duplicate_genes_to_drop = [\n",
    "    \"fig|106370.16.peg.4543\",\n",
    "    \"fig|111802.3.peg.9038\",\n",
    "    \"fig|1120935.3.peg.6134\",\n",
    "    \"fig|1123023.3.peg.5886\",\n",
    "    \"fig|1202450.10.peg.2450\",\n",
    "    \"fig|1206731.4.peg.7885\",\n",
    "    \"fig|1206741.4.peg.4760\",\n",
    "    \"fig|1210062.4.peg.6350\",\n",
    "    \"fig|1210064.5.peg.5593\",\n",
    "    \"fig|1210084.4.peg.9515\",\n",
    "    \"fig|1220561.3.peg.4400\",\n",
    "    \"fig|1220564.3.peg.6715\",\n",
    "    \"fig|1238180.7.peg.744\",\n",
    "    \"fig|1265311.3.peg.2802\",\n",
    "    \"fig|1490222.3.peg.8831\",\n",
    "    \"fig|1719.1356.peg.1786\"\n",
    "    ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d68ad89c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3580d2bc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "5fa74035",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    466.000000\n",
       "mean      80.894691\n",
       "std        3.591385\n",
       "min       67.460000\n",
       "25%       78.934250\n",
       "50%       81.103000\n",
       "75%       83.210000\n",
       "max       92.944000\n",
       "Name: pident, dtype: float64"
      ]
     },
     "execution_count": 263,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_dep['pident'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "5370793e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# bsap_filter_df[bsap_filter_df['Genome Name'].isin(biob_dep[biob_dep['Genome Name'].duplicated(keep=False)]['Genome Name'].unique())][\n",
    "#     ['seqid', 'bitscore', 'pident', 'length', 'sstart', 'ssend', 'Genome Name', 'feature.genome_id', 'feature.patric_id']].reset_index(drop=True).loc[30:41]\n",
    "\n",
    "# biob_dep[biob_dep['Genome Name'].duplicated(keep=False)][\n",
    "#     ['seqid', 'bitscore', 'pident', 'length', 'sstart', 'ssend', 'Genome Name', 'feature.genome_id', 'feature.patric_id']].reset_index(drop=True).loc[30:41]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "ccb122c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_dep = biob_dep[~biob_dep['feature.patric_id'].isin(duplicate_genes_to_drop)].dropna().drop_duplicates(subset=['Genome ID']).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "id": "4d7d1a33",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>...</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>NCBI Taxon ID</th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>feature.genome_id</th>\n",
       "      <th>feature.patric_id</th>\n",
       "      <th>feature.na_sequence</th>\n",
       "      <th>feature.aa_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [seqid, evalue, bitscore, pident, length, staxid, ssname, scomname, sstart, ssend, sseq, Genome Name, NCBI Taxon ID, Genome ID, Order, Family, Genus, feature.genome_id, feature.patric_id, feature.na_sequence, feature.aa_sequence]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 21 columns]"
      ]
     },
     "execution_count": 266,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_dep[biob_dep['Genome Name'].duplicated(keep=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "e897697a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(425, 21)"
      ]
     },
     "execution_count": 267,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_dep.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "2923221a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>...</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>NCBI Taxon ID</th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>feature.genome_id</th>\n",
       "      <th>feature.patric_id</th>\n",
       "      <th>feature.na_sequence</th>\n",
       "      <th>feature.aa_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gi|1360579553|ref|NZ_CP027433.1|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>836.0</td>\n",
       "      <td>78.926</td>\n",
       "      <td>987</td>\n",
       "      <td>1004901</td>\n",
       "      <td>Gordonia iterans</td>\n",
       "      <td>Gordonia iterans</td>\n",
       "      <td>2728595</td>\n",
       "      <td>2727610</td>\n",
       "      <td>...</td>\n",
       "      <td>Gordonia iterans strain Co17</td>\n",
       "      <td>1004901</td>\n",
       "      <td>1004901.30</td>\n",
       "      <td>Corynebacteriales</td>\n",
       "      <td>Gordoniaceae</td>\n",
       "      <td>Gordonia</td>\n",
       "      <td>1004901.30</td>\n",
       "      <td>fig|1004901.3.peg.2573</td>\n",
       "      <td>gtgaccaccgtcgacgaggcccccgcggccgccaccgaccagaccg...</td>\n",
       "      <td>MTTVDEAPAAATDQTESDILVVAREQVLECGEALRYEQILDVLRLP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gi|400537151|ref|NZ_AFVW02000007.1|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1290.0</td>\n",
       "      <td>87.226</td>\n",
       "      <td>1049</td>\n",
       "      <td>1041522</td>\n",
       "      <td>Mycobacterium colombiense CECT 3035</td>\n",
       "      <td>Mycobacterium colombiense CECT 3035</td>\n",
       "      <td>261638</td>\n",
       "      <td>260602</td>\n",
       "      <td>...</td>\n",
       "      <td>Mycobacterium colombiense CECT 3035</td>\n",
       "      <td>1041522</td>\n",
       "      <td>1041522.30</td>\n",
       "      <td>Corynebacteriales</td>\n",
       "      <td>Mycobacteriaceae</td>\n",
       "      <td>Mycobacterium</td>\n",
       "      <td>1041522.30</td>\n",
       "      <td>fig|1041522.3.peg.3658</td>\n",
       "      <td>gtgactcaagcggcgacgcgaccgacggccgaagccggcagcgacg...</td>\n",
       "      <td>MTQAATRPTAEAGSDEDILAVARQQVLQDGQGLSRDQVLRVLQLPD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gi|2228921663|ref|NZ_JALKBW010000001.1|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>884.0</td>\n",
       "      <td>79.838</td>\n",
       "      <td>987</td>\n",
       "      <td>1053547</td>\n",
       "      <td>Gordonia alkaliphila</td>\n",
       "      <td>Gordonia alkaliphila</td>\n",
       "      <td>3497259</td>\n",
       "      <td>3496273</td>\n",
       "      <td>...</td>\n",
       "      <td>Gordonia alkaliphila WW102</td>\n",
       "      <td>1053547</td>\n",
       "      <td>1053547.23</td>\n",
       "      <td>Corynebacteriales</td>\n",
       "      <td>Gordoniaceae</td>\n",
       "      <td>Gordonia</td>\n",
       "      <td>1053547.23</td>\n",
       "      <td>fig|1053547.23.peg.3220</td>\n",
       "      <td>atgaccatcaccaccgcctccccgtcgatcgacgacaccgccgccg...</td>\n",
       "      <td>MTITTASPSIDDTAADRDILDVARTQVLDEGVGLNYDQLLQVLNLD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gi|86738724|ref|NC_007777.1|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>646.0</td>\n",
       "      <td>75.368</td>\n",
       "      <td>950</td>\n",
       "      <td>106370</td>\n",
       "      <td>Frankia casuarinae</td>\n",
       "      <td>Frankia casuarinae</td>\n",
       "      <td>4512150</td>\n",
       "      <td>4513092</td>\n",
       "      <td>...</td>\n",
       "      <td>Frankia sp. CcI3</td>\n",
       "      <td>106370</td>\n",
       "      <td>106370.16</td>\n",
       "      <td>Frankiales</td>\n",
       "      <td>Frankiaceae</td>\n",
       "      <td>Frankia</td>\n",
       "      <td>106370.16</td>\n",
       "      <td>fig|106370.16.peg.4098</td>\n",
       "      <td>gtgctcgccgtgctccggctcccggacgagacgctgaccgatctgc...</td>\n",
       "      <td>MLAVLRLPDETLTDLLALAHEVRMRWCGPEVEVEGIVSLKTGGCPE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gi|487404592|ref|NZ_ARVW01000001.1|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>782.0</td>\n",
       "      <td>78.452</td>\n",
       "      <td>956</td>\n",
       "      <td>1068980</td>\n",
       "      <td>Amycolatopsis nigrescens CSC17Ta-90</td>\n",
       "      <td>Amycolatopsis nigrescens CSC17Ta-90</td>\n",
       "      <td>5988697</td>\n",
       "      <td>5989649</td>\n",
       "      <td>...</td>\n",
       "      <td>Amycolatopsis nigrescens CSC17Ta-90</td>\n",
       "      <td>1068980</td>\n",
       "      <td>1068980.30</td>\n",
       "      <td>Pseudonocardiales</td>\n",
       "      <td>Pseudonocardiaceae</td>\n",
       "      <td>Amycolatopsis</td>\n",
       "      <td>1068980.30</td>\n",
       "      <td>fig|1068980.3.peg.5690</td>\n",
       "      <td>gtgaccgcagcacccggacaggccgagcaggtccaccgcgccgacg...</td>\n",
       "      <td>MTAAPGQAEQVHRADADVLAVAREQVLERGTGLGEQQLLEVLRLGD...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     seqid  evalue  bitscore  pident  length  \\\n",
       "0         gi|1360579553|ref|NZ_CP027433.1|     0.0     836.0  78.926     987   \n",
       "1      gi|400537151|ref|NZ_AFVW02000007.1|     0.0    1290.0  87.226    1049   \n",
       "2  gi|2228921663|ref|NZ_JALKBW010000001.1|     0.0     884.0  79.838     987   \n",
       "3             gi|86738724|ref|NC_007777.1|     0.0     646.0  75.368     950   \n",
       "4      gi|487404592|ref|NZ_ARVW01000001.1|     0.0     782.0  78.452     956   \n",
       "\n",
       "    staxid                               ssname  \\\n",
       "0  1004901                     Gordonia iterans   \n",
       "1  1041522  Mycobacterium colombiense CECT 3035   \n",
       "2  1053547                 Gordonia alkaliphila   \n",
       "3   106370                   Frankia casuarinae   \n",
       "4  1068980  Amycolatopsis nigrescens CSC17Ta-90   \n",
       "\n",
       "                              scomname   sstart    ssend  ...  \\\n",
       "0                     Gordonia iterans  2728595  2727610  ...   \n",
       "1  Mycobacterium colombiense CECT 3035   261638   260602  ...   \n",
       "2                 Gordonia alkaliphila  3497259  3496273  ...   \n",
       "3                   Frankia casuarinae  4512150  4513092  ...   \n",
       "4  Amycolatopsis nigrescens CSC17Ta-90  5988697  5989649  ...   \n",
       "\n",
       "                           Genome Name NCBI Taxon ID   Genome ID  \\\n",
       "0         Gordonia iterans strain Co17       1004901  1004901.30   \n",
       "1  Mycobacterium colombiense CECT 3035       1041522  1041522.30   \n",
       "2           Gordonia alkaliphila WW102       1053547  1053547.23   \n",
       "3                     Frankia sp. CcI3        106370   106370.16   \n",
       "4  Amycolatopsis nigrescens CSC17Ta-90       1068980  1068980.30   \n",
       "\n",
       "               Order              Family          Genus feature.genome_id  \\\n",
       "0  Corynebacteriales        Gordoniaceae       Gordonia        1004901.30   \n",
       "1  Corynebacteriales    Mycobacteriaceae  Mycobacterium        1041522.30   \n",
       "2  Corynebacteriales        Gordoniaceae       Gordonia        1053547.23   \n",
       "3         Frankiales         Frankiaceae        Frankia         106370.16   \n",
       "4  Pseudonocardiales  Pseudonocardiaceae  Amycolatopsis        1068980.30   \n",
       "\n",
       "         feature.patric_id                                feature.na_sequence  \\\n",
       "0   fig|1004901.3.peg.2573  gtgaccaccgtcgacgaggcccccgcggccgccaccgaccagaccg...   \n",
       "1   fig|1041522.3.peg.3658  gtgactcaagcggcgacgcgaccgacggccgaagccggcagcgacg...   \n",
       "2  fig|1053547.23.peg.3220  atgaccatcaccaccgcctccccgtcgatcgacgacaccgccgccg...   \n",
       "3   fig|106370.16.peg.4098  gtgctcgccgtgctccggctcccggacgagacgctgaccgatctgc...   \n",
       "4   fig|1068980.3.peg.5690  gtgaccgcagcacccggacaggccgagcaggtccaccgcgccgacg...   \n",
       "\n",
       "                                 feature.aa_sequence  \n",
       "0  MTTVDEAPAAATDQTESDILVVAREQVLECGEALRYEQILDVLRLP...  \n",
       "1  MTQAATRPTAEAGSDEDILAVARQQVLQDGQGLSRDQVLRVLQLPD...  \n",
       "2  MTITTASPSIDDTAADRDILDVARTQVLDEGVGLNYDQLLQVLNLD...  \n",
       "3  MLAVLRLPDETLTDLLALAHEVRMRWCGPEVEVEGIVSLKTGGCPE...  \n",
       "4  MTAAPGQAEQVHRADADVLAVAREQVLERGTGLGEQQLLEVLRLGD...  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 268,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_dep.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "862ad5b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count     425.000000\n",
       "mean     1016.188235\n",
       "std        57.653506\n",
       "min       189.000000\n",
       "25%      1005.000000\n",
       "50%      1014.000000\n",
       "75%      1038.000000\n",
       "max      1101.000000\n",
       "Name: feature.na_sequence, dtype: float64"
      ]
     },
     "execution_count": 269,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_dep['feature.na_sequence'].apply(len).describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "ec5b2565",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_dep = biob_dep[biob_dep['feature.na_sequence'].apply(len) < 1050]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fbf958b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "0ff6560c",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_dep[subset_cols_of_interest].to_csv(\"motif/biob_dependent_seqs.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "6cebf5da",
   "metadata": {},
   "outputs": [],
   "source": [
    "# write_df_to_fasta(biob_dep, 'feature.na_sequence', 'Genome Name', \"motif/putative2_biob_dependent_seqs_nt.fasta\")\n",
    "write_df_to_fasta(biob_dep, 'feature.aa_sequence', 'Genome Name', \"motif/putative2_biob_dependent_seqs_aa_filter3.fasta\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61ba5514",
   "metadata": {},
   "source": [
    "## hack together an alternate tremmer lm file to drop out the long biob copies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "8fdb2c10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(344, 21)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "28ede279",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count     344.000000\n",
       "mean     1013.843023\n",
       "std        48.426420\n",
       "min       477.000000\n",
       "25%       996.000000\n",
       "50%       996.000000\n",
       "75%      1020.000000\n",
       "max      1194.000000\n",
       "Name: feature.na_sequence, dtype: float64"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_subset['feature.na_sequence'].apply(len).describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "31321f43",
   "metadata": {},
   "outputs": [],
   "source": [
    "lm = pd.read_csv(\"treemmer/biob_family_lm.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "c324541d",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_ind_lm_edit = biob_subset[['Genome Name', 'Family']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "55e79643",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/nataliethornton/anaconda3/envs/ete/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "biob_ind_lm_edit['name'] = biob_ind_lm_edit['Genome Name'].str.replace(\" \", \"_\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "c9c80db2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Family</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Nonomuraea soli strain DSM 45533</td>\n",
       "      <td>Streptosporangiaceae</td>\n",
       "      <td>Nonomuraea_soli_strain_DSM_45533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Actinoplanes atraurantiacus strain CGMCC 4.6857</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Actinoplanes_atraurantiacus_strain_CGMCC_4.6857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Kutzneria kofuensis strain DSM 43851</td>\n",
       "      <td>Pseudonocardiaceae</td>\n",
       "      <td>Kutzneria_kofuensis_strain_DSM_43851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Actinoplanes rishiriensis strain NBRC 108556</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Actinoplanes_rishiriensis_strain_NBRC_108556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Salinispora pacifica DSM 45546</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Salinispora_pacifica_DSM_45546</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Genome Name                Family  \\\n",
       "0                 Nonomuraea soli strain DSM 45533  Streptosporangiaceae   \n",
       "1  Actinoplanes atraurantiacus strain CGMCC 4.6857    Micromonosporaceae   \n",
       "2             Kutzneria kofuensis strain DSM 43851    Pseudonocardiaceae   \n",
       "3     Actinoplanes rishiriensis strain NBRC 108556    Micromonosporaceae   \n",
       "4                   Salinispora pacifica DSM 45546    Micromonosporaceae   \n",
       "\n",
       "                                              name  \n",
       "0                 Nonomuraea_soli_strain_DSM_45533  \n",
       "1  Actinoplanes_atraurantiacus_strain_CGMCC_4.6857  \n",
       "2             Kutzneria_kofuensis_strain_DSM_43851  \n",
       "3     Actinoplanes_rishiriensis_strain_NBRC_108556  \n",
       "4                   Salinispora_pacifica_DSM_45546  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_ind_lm_edit.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "ae01b6c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_ind_lm_edit = lm.merge(biob_ind_lm_edit, how='left', on='name', suffixes=('_old', ''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "c7d9c313",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_ind_lm_edit = biob_ind_lm_edit[['name', 'Family']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "36c1d980",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/nataliethornton/anaconda3/envs/ete/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "biob_ind_lm_edit['Family'] = biob_ind_lm_edit['Family'].fillna('drop')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "e3f0f8a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_ind_lm_edit.to_csv('treemmer/biob_family_lm_independentEDIT.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "6fb26ac4",
   "metadata": {},
   "outputs": [],
   "source": [
    "indlist = pd.read_csv(\"motif/biob_independent_full_consensus_tree.nwk_trimmed_list_X_100\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "00c486a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(100, 1)"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indlist.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "f3a54542",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biob_ind_lm_edit.merge(indlist, left_on='name', right_on=0)['Family'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c4d5df5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65c89d4e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "231a3822",
   "metadata": {},
   "source": [
    "## read in aa msas and generate base frequency matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "e89a0e99",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ind = read_in_fasta(\"motif/putative2_biob_independent_seqs_aa.mfa\")\n",
    "# dep = read_in_fasta(\"motif/putative2_biob_dependent_seqs_aa.mfa\")\n",
    "\n",
    "# ind = read_in_fasta(\"motif/putative2_biob_independent_seqs_nt.mfa\")\n",
    "# dep = read_in_fasta(\"motif/putative2_biob_dependent_seqs_nt.mfa\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "fab2cff1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dep_seq = pd.DataFrame(dep['seq'].apply(str).apply(lambda x: list(itertools.chain.from_iterable(x))).tolist(), index=dep['id'])\n",
    "# ind_seq = pd.DataFrame(ind['seq'].apply(str).apply(lambda x: list(itertools.chain.from_iterable(x))).tolist(), index=ind['id'])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e3269c9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "biob_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45da00da",
   "metadata": {},
   "outputs": [],
   "source": [
    "# write_df_to_fasta(biob_dep, 'feature.na_sequence', 'Genome Name', \"motif/putative2_biob_dependent_seqs_nt.fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "id": "f65e746e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dep_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "795e1ed5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ind_seq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4297c04e",
   "metadata": {},
   "source": [
    "## write Tremmer subset100 independent and dependent biob copies to a merged fasta:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "dc67b812",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ind = pd.read_csv(\"motif/biob_independent_full_consensus_tree.nwk_trimmed_list_X_100\", header=None)\n",
    "\n",
    "# ind = pd.read_csv(\"motif/biob_independent_full_edit_consensus_tree.nwk_trimmed_list_X_100\", header=None)\n",
    "# dep = pd.read_csv(\"motif/biob_dependent_full_consensus_tree.nwk_trimmed_list_X_100\", header=None)\n",
    "\n",
    "ind = pd.read_csv(\"motif/putative2_biob_independent_seqs_aa_filter3_consensus.nwk_trimmed_list_X_100\", header=None)\n",
    "dep = pd.read_csv(\"motif/putative2_biob_dependent_seqs_aa_filter3_consensus.nwk_trimmed_list_X_100\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "0125286c",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df_ind = biob_subset[biob_subset['Genome Name'].isin(ind[0].str.replace(\"_\", \" \"))].copy()\n",
    "merged_df_dep = biob_dep[biob_dep['Genome Name'].isin(dep[0].str.replace(\"_\", \" \"))].copy()\n",
    "\n",
    "merged_df_ind['type'] = 'ind'\n",
    "merged_df_dep['type'] = 'dep'\n",
    "\n",
    "merged_df = pd.concat([merged_df_ind, merged_df_dep]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "4dab4c09",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>...</th>\n",
       "      <th>NCBI Taxon ID</th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>feature.genome_id</th>\n",
       "      <th>feature.patric_id</th>\n",
       "      <th>feature.na_sequence</th>\n",
       "      <th>feature.aa_sequence</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gi|1884111766|ref|NZ_JACDUR010000005.1|</td>\n",
       "      <td>1.470000e-180</td>\n",
       "      <td>631.0</td>\n",
       "      <td>75.362</td>\n",
       "      <td>966</td>\n",
       "      <td>1032476</td>\n",
       "      <td>Nonomuraea soli</td>\n",
       "      <td>Nonomuraea soli</td>\n",
       "      <td>1028404</td>\n",
       "      <td>1029363</td>\n",
       "      <td>...</td>\n",
       "      <td>1032476</td>\n",
       "      <td>1032476.3</td>\n",
       "      <td>Streptosporangiales</td>\n",
       "      <td>Streptosporangiaceae</td>\n",
       "      <td>Nonomuraea</td>\n",
       "      <td>1032476.3</td>\n",
       "      <td>fig|1032476.3.peg.5624</td>\n",
       "      <td>gtgacgatcctcgacatcgcccgcacccaggtcctggagcagggca...</td>\n",
       "      <td>MTILDIARTQVLEQGKGLDREQALRCLLLDDALLPDLLALAHEVRM...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gi|1254547583|ref|NZ_OBDY01000010.1|</td>\n",
       "      <td>1.220000e-175</td>\n",
       "      <td>614.0</td>\n",
       "      <td>74.920</td>\n",
       "      <td>937</td>\n",
       "      <td>1036182</td>\n",
       "      <td>Actinoplanes atraurantiacus</td>\n",
       "      <td>Actinoplanes atraurantiacus</td>\n",
       "      <td>77996</td>\n",
       "      <td>77063</td>\n",
       "      <td>...</td>\n",
       "      <td>1036182</td>\n",
       "      <td>1036182.3</td>\n",
       "      <td>Micromonosporales</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Actinoplanes</td>\n",
       "      <td>1036182.3</td>\n",
       "      <td>fig|1036182.3.peg.2302</td>\n",
       "      <td>atgcccgagatcctcgagatggcccgtgcccaggttctcgccggcg...</td>\n",
       "      <td>MPEILEMARAQVLAGGAGLGEAQILEVLRLGDDHLPELLQLAHDVR...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gi|1892753580|ref|NZ_JACHIR010000001.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>753.0</td>\n",
       "      <td>77.409</td>\n",
       "      <td>965</td>\n",
       "      <td>103725</td>\n",
       "      <td>Kutzneria kofuensis</td>\n",
       "      <td>Kutzneria kofuensis</td>\n",
       "      <td>5947414</td>\n",
       "      <td>5948375</td>\n",
       "      <td>...</td>\n",
       "      <td>103725</td>\n",
       "      <td>103725.4</td>\n",
       "      <td>Pseudonocardiales</td>\n",
       "      <td>Pseudonocardiaceae</td>\n",
       "      <td>Kutzneria</td>\n",
       "      <td>103725.4</td>\n",
       "      <td>fig|103725.4.peg.5640</td>\n",
       "      <td>gtgaccgccagcgtggacaccgacatcatcgccgtcgcccgtgagc...</td>\n",
       "      <td>MTASVDTDIIAVAREQVLERGEGLSQEQLLQVLKLGDDRLAELLQL...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gi|1892940858|ref|NZ_JACHGN010000004.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>725.0</td>\n",
       "      <td>76.737</td>\n",
       "      <td>993</td>\n",
       "      <td>1073253</td>\n",
       "      <td>Thermocatellispora tengchongensis</td>\n",
       "      <td>Thermocatellispora tengchongensis</td>\n",
       "      <td>576796</td>\n",
       "      <td>577782</td>\n",
       "      <td>...</td>\n",
       "      <td>1073253</td>\n",
       "      <td>1073253.3</td>\n",
       "      <td>Streptosporangiales</td>\n",
       "      <td>Streptosporangiaceae</td>\n",
       "      <td>Thermocatellispora</td>\n",
       "      <td>1073253.3</td>\n",
       "      <td>fig|1073253.3.peg.2482</td>\n",
       "      <td>gtggacattctgaccgtcgcccgcgaccaggtgctcgaccggggcg...</td>\n",
       "      <td>MDILTVARDQVLDRGEGLTAEQALECLRLPDERLPELLALAHEVRV...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gi|1850378389|ref|NZ_AP022870.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>655.0</td>\n",
       "      <td>75.602</td>\n",
       "      <td>955</td>\n",
       "      <td>1076124</td>\n",
       "      <td>Phytohabitans flavus</td>\n",
       "      <td>Phytohabitans flavus</td>\n",
       "      <td>9487101</td>\n",
       "      <td>9486152</td>\n",
       "      <td>...</td>\n",
       "      <td>1076124</td>\n",
       "      <td>1076124.3</td>\n",
       "      <td>Micromonosporales</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Phytohabitans</td>\n",
       "      <td>1076124.3</td>\n",
       "      <td>fig|1076124.3.peg.9236</td>\n",
       "      <td>atgccagagatcctcgacatcgctcgtgcccaggtgctcgaacgcg...</td>\n",
       "      <td>MPEILDIARAQVLERGTGLDEAGVLAVLRLPDEHVPDALQLAHEVR...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     seqid         evalue  bitscore  pident  \\\n",
       "0  gi|1884111766|ref|NZ_JACDUR010000005.1|  1.470000e-180     631.0  75.362   \n",
       "1     gi|1254547583|ref|NZ_OBDY01000010.1|  1.220000e-175     614.0  74.920   \n",
       "2  gi|1892753580|ref|NZ_JACHIR010000001.1|   0.000000e+00     753.0  77.409   \n",
       "3  gi|1892940858|ref|NZ_JACHGN010000004.1|   0.000000e+00     725.0  76.737   \n",
       "4         gi|1850378389|ref|NZ_AP022870.1|   0.000000e+00     655.0  75.602   \n",
       "\n",
       "   length   staxid                             ssname  \\\n",
       "0     966  1032476                    Nonomuraea soli   \n",
       "1     937  1036182        Actinoplanes atraurantiacus   \n",
       "2     965   103725                Kutzneria kofuensis   \n",
       "3     993  1073253  Thermocatellispora tengchongensis   \n",
       "4     955  1076124               Phytohabitans flavus   \n",
       "\n",
       "                            scomname   sstart    ssend  ... NCBI Taxon ID  \\\n",
       "0                    Nonomuraea soli  1028404  1029363  ...       1032476   \n",
       "1        Actinoplanes atraurantiacus    77996    77063  ...       1036182   \n",
       "2                Kutzneria kofuensis  5947414  5948375  ...        103725   \n",
       "3  Thermocatellispora tengchongensis   576796   577782  ...       1073253   \n",
       "4               Phytohabitans flavus  9487101  9486152  ...       1076124   \n",
       "\n",
       "   Genome ID                Order                Family               Genus  \\\n",
       "0  1032476.3  Streptosporangiales  Streptosporangiaceae          Nonomuraea   \n",
       "1  1036182.3    Micromonosporales    Micromonosporaceae        Actinoplanes   \n",
       "2   103725.4    Pseudonocardiales    Pseudonocardiaceae           Kutzneria   \n",
       "3  1073253.3  Streptosporangiales  Streptosporangiaceae  Thermocatellispora   \n",
       "4  1076124.3    Micromonosporales    Micromonosporaceae       Phytohabitans   \n",
       "\n",
       "  feature.genome_id       feature.patric_id  \\\n",
       "0         1032476.3  fig|1032476.3.peg.5624   \n",
       "1         1036182.3  fig|1036182.3.peg.2302   \n",
       "2          103725.4   fig|103725.4.peg.5640   \n",
       "3         1073253.3  fig|1073253.3.peg.2482   \n",
       "4         1076124.3  fig|1076124.3.peg.9236   \n",
       "\n",
       "                                 feature.na_sequence  \\\n",
       "0  gtgacgatcctcgacatcgcccgcacccaggtcctggagcagggca...   \n",
       "1  atgcccgagatcctcgagatggcccgtgcccaggttctcgccggcg...   \n",
       "2  gtgaccgccagcgtggacaccgacatcatcgccgtcgcccgtgagc...   \n",
       "3  gtggacattctgaccgtcgcccgcgaccaggtgctcgaccggggcg...   \n",
       "4  atgccagagatcctcgacatcgctcgtgcccaggtgctcgaacgcg...   \n",
       "\n",
       "                                 feature.aa_sequence type  \n",
       "0  MTILDIARTQVLEQGKGLDREQALRCLLLDDALLPDLLALAHEVRM...  ind  \n",
       "1  MPEILEMARAQVLAGGAGLGEAQILEVLRLGDDHLPELLQLAHDVR...  ind  \n",
       "2  MTASVDTDIIAVAREQVLERGEGLSQEQLLQVLKLGDDRLAELLQL...  ind  \n",
       "3  MDILTVARDQVLDRGEGLTAEQALECLRLPDERLPELLALAHEVRV...  ind  \n",
       "4  MPEILDIARAQVLERGTGLDEAGVLAVLRLPDEHVPDALQLAHEVR...  ind  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "4bfdc65b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seqid', 'evalue', 'bitscore', 'pident', 'length', 'staxid', 'ssname',\n",
       "       'scomname', 'sstart', 'ssend', 'sseq', 'Genome Name', 'NCBI Taxon ID',\n",
       "       'Genome ID', 'Order', 'Family', 'Genus', 'feature.genome_id',\n",
       "       'feature.patric_id', 'feature.na_sequence', 'feature.aa_sequence',\n",
       "       'type'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 291,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "7abb0396",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>staxid</th>\n",
       "      <th>ssname</th>\n",
       "      <th>scomname</th>\n",
       "      <th>sstart</th>\n",
       "      <th>ssend</th>\n",
       "      <th>...</th>\n",
       "      <th>NCBI Taxon ID</th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>feature.genome_id</th>\n",
       "      <th>feature.patric_id</th>\n",
       "      <th>feature.na_sequence</th>\n",
       "      <th>feature.aa_sequence</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gi|1884111766|ref|NZ_JACDUR010000005.1|</td>\n",
       "      <td>1.470000e-180</td>\n",
       "      <td>631.0</td>\n",
       "      <td>75.362</td>\n",
       "      <td>966</td>\n",
       "      <td>1032476</td>\n",
       "      <td>Nonomuraea soli</td>\n",
       "      <td>Nonomuraea soli</td>\n",
       "      <td>1028404</td>\n",
       "      <td>1029363</td>\n",
       "      <td>...</td>\n",
       "      <td>1032476</td>\n",
       "      <td>1032476.3</td>\n",
       "      <td>Streptosporangiales</td>\n",
       "      <td>Streptosporangiaceae</td>\n",
       "      <td>Nonomuraea</td>\n",
       "      <td>1032476.3</td>\n",
       "      <td>fig|1032476.3.peg.5624</td>\n",
       "      <td>gtgacgatcctcgacatcgcccgcacccaggtcctggagcagggca...</td>\n",
       "      <td>MTILDIARTQVLEQGKGLDREQALRCLLLDDALLPDLLALAHEVRM...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gi|1254547583|ref|NZ_OBDY01000010.1|</td>\n",
       "      <td>1.220000e-175</td>\n",
       "      <td>614.0</td>\n",
       "      <td>74.920</td>\n",
       "      <td>937</td>\n",
       "      <td>1036182</td>\n",
       "      <td>Actinoplanes atraurantiacus</td>\n",
       "      <td>Actinoplanes atraurantiacus</td>\n",
       "      <td>77996</td>\n",
       "      <td>77063</td>\n",
       "      <td>...</td>\n",
       "      <td>1036182</td>\n",
       "      <td>1036182.3</td>\n",
       "      <td>Micromonosporales</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Actinoplanes</td>\n",
       "      <td>1036182.3</td>\n",
       "      <td>fig|1036182.3.peg.2302</td>\n",
       "      <td>atgcccgagatcctcgagatggcccgtgcccaggttctcgccggcg...</td>\n",
       "      <td>MPEILEMARAQVLAGGAGLGEAQILEVLRLGDDHLPELLQLAHDVR...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gi|1892753580|ref|NZ_JACHIR010000001.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>753.0</td>\n",
       "      <td>77.409</td>\n",
       "      <td>965</td>\n",
       "      <td>103725</td>\n",
       "      <td>Kutzneria kofuensis</td>\n",
       "      <td>Kutzneria kofuensis</td>\n",
       "      <td>5947414</td>\n",
       "      <td>5948375</td>\n",
       "      <td>...</td>\n",
       "      <td>103725</td>\n",
       "      <td>103725.4</td>\n",
       "      <td>Pseudonocardiales</td>\n",
       "      <td>Pseudonocardiaceae</td>\n",
       "      <td>Kutzneria</td>\n",
       "      <td>103725.4</td>\n",
       "      <td>fig|103725.4.peg.5640</td>\n",
       "      <td>gtgaccgccagcgtggacaccgacatcatcgccgtcgcccgtgagc...</td>\n",
       "      <td>MTASVDTDIIAVAREQVLERGEGLSQEQLLQVLKLGDDRLAELLQL...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gi|1892940858|ref|NZ_JACHGN010000004.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>725.0</td>\n",
       "      <td>76.737</td>\n",
       "      <td>993</td>\n",
       "      <td>1073253</td>\n",
       "      <td>Thermocatellispora tengchongensis</td>\n",
       "      <td>Thermocatellispora tengchongensis</td>\n",
       "      <td>576796</td>\n",
       "      <td>577782</td>\n",
       "      <td>...</td>\n",
       "      <td>1073253</td>\n",
       "      <td>1073253.3</td>\n",
       "      <td>Streptosporangiales</td>\n",
       "      <td>Streptosporangiaceae</td>\n",
       "      <td>Thermocatellispora</td>\n",
       "      <td>1073253.3</td>\n",
       "      <td>fig|1073253.3.peg.2482</td>\n",
       "      <td>gtggacattctgaccgtcgcccgcgaccaggtgctcgaccggggcg...</td>\n",
       "      <td>MDILTVARDQVLDRGEGLTAEQALECLRLPDERLPELLALAHEVRV...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gi|1850378389|ref|NZ_AP022870.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>655.0</td>\n",
       "      <td>75.602</td>\n",
       "      <td>955</td>\n",
       "      <td>1076124</td>\n",
       "      <td>Phytohabitans flavus</td>\n",
       "      <td>Phytohabitans flavus</td>\n",
       "      <td>9487101</td>\n",
       "      <td>9486152</td>\n",
       "      <td>...</td>\n",
       "      <td>1076124</td>\n",
       "      <td>1076124.3</td>\n",
       "      <td>Micromonosporales</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Phytohabitans</td>\n",
       "      <td>1076124.3</td>\n",
       "      <td>fig|1076124.3.peg.9236</td>\n",
       "      <td>atgccagagatcctcgacatcgctcgtgcccaggtgctcgaacgcg...</td>\n",
       "      <td>MPEILDIARAQVLERGTGLDEAGVLAVLRLPDEHVPDALQLAHEVR...</td>\n",
       "      <td>ind</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>gi|2175968702|ref|NZ_CP090063.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>744.0</td>\n",
       "      <td>77.503</td>\n",
       "      <td>969</td>\n",
       "      <td>715473</td>\n",
       "      <td>Amycolatopsis acidiphila</td>\n",
       "      <td>Amycolatopsis acidiphila</td>\n",
       "      <td>6252312</td>\n",
       "      <td>6251348</td>\n",
       "      <td>...</td>\n",
       "      <td>715473</td>\n",
       "      <td>715473.5</td>\n",
       "      <td>Pseudonocardiales</td>\n",
       "      <td>Pseudonocardiaceae</td>\n",
       "      <td>Amycolatopsis</td>\n",
       "      <td>715473.5</td>\n",
       "      <td>fig|715473.5.peg.6308</td>\n",
       "      <td>gtgacctcagcacccgaacagaccgacgtggtcgccgtcgcccggg...</td>\n",
       "      <td>MTSAPEQTDVVAVAREQVLEQGIGLSQRQVLDVLRLADDRLSELLA...</td>\n",
       "      <td>dep</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>gi|1697805133|ref|NZ_BJNG01000006.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>866.0</td>\n",
       "      <td>80.144</td>\n",
       "      <td>972</td>\n",
       "      <td>76726</td>\n",
       "      <td>Pseudonocardia hydrocarbonoxydans</td>\n",
       "      <td>Pseudonocardia hydrocarbonoxydans</td>\n",
       "      <td>46003</td>\n",
       "      <td>45037</td>\n",
       "      <td>...</td>\n",
       "      <td>76726</td>\n",
       "      <td>76726.4</td>\n",
       "      <td>Pseudonocardiales</td>\n",
       "      <td>Pseudonocardiaceae</td>\n",
       "      <td>Pseudonocardia</td>\n",
       "      <td>76726.4</td>\n",
       "      <td>fig|76726.4.peg.840</td>\n",
       "      <td>atggccgaagccgccgccatcctcgccaccgcccgcacccgggtgc...</td>\n",
       "      <td>MAEAAAILATARTRVLEEGVGLDEAQVLEVLRLPDEALDDLLQLAH...</td>\n",
       "      <td>dep</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>gi|1986697868|ref|NZ_BOOC01000053.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>673.0</td>\n",
       "      <td>75.635</td>\n",
       "      <td>985</td>\n",
       "      <td>83302</td>\n",
       "      <td>Microbispora corallina</td>\n",
       "      <td>Microbispora corallina</td>\n",
       "      <td>20894</td>\n",
       "      <td>21873</td>\n",
       "      <td>...</td>\n",
       "      <td>83302</td>\n",
       "      <td>83302.3</td>\n",
       "      <td>Streptosporangiales</td>\n",
       "      <td>Streptosporangiaceae</td>\n",
       "      <td>Microbispora</td>\n",
       "      <td>83302.3</td>\n",
       "      <td>fig|83302.3.peg.7395</td>\n",
       "      <td>gtgaacactgacatcgtggagatcgcccgcgtccaggtgctcgagg...</td>\n",
       "      <td>MNTDIVEIARVQVLEEGRGLDAAQALECLTLPDDRLPELLALAHEV...</td>\n",
       "      <td>dep</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>gi|1816837986|ref|NZ_BLKW01000004.1|</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1098.0</td>\n",
       "      <td>84.438</td>\n",
       "      <td>996</td>\n",
       "      <td>84962</td>\n",
       "      <td>Mycobacterium botniense</td>\n",
       "      <td>Mycobacterium botniense</td>\n",
       "      <td>1033100</td>\n",
       "      <td>1034095</td>\n",
       "      <td>...</td>\n",
       "      <td>84962</td>\n",
       "      <td>84962.6</td>\n",
       "      <td>Corynebacteriales</td>\n",
       "      <td>Mycobacteriaceae</td>\n",
       "      <td>Mycobacterium</td>\n",
       "      <td>84962.6</td>\n",
       "      <td>fig|84962.6.peg.3194</td>\n",
       "      <td>gtgacgcaggcgatgacccggccggccgccgatgacagcaacaccg...</td>\n",
       "      <td>MTQAMTRPAADDSNTDVLAVARRQVLELGKGLTRDQVLQVLQLPDE...</td>\n",
       "      <td>dep</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>gi|924867397|ref|NZ_CP009220.1|</td>\n",
       "      <td>2.010000e-128</td>\n",
       "      <td>456.0</td>\n",
       "      <td>71.109</td>\n",
       "      <td>938</td>\n",
       "      <td>931089</td>\n",
       "      <td>Corynebacterium deserti GIMN1.010</td>\n",
       "      <td>Corynebacterium deserti GIMN1.010</td>\n",
       "      <td>68759</td>\n",
       "      <td>69694</td>\n",
       "      <td>...</td>\n",
       "      <td>931089</td>\n",
       "      <td>931089.4</td>\n",
       "      <td>Corynebacteriales</td>\n",
       "      <td>Corynebacteriaceae</td>\n",
       "      <td>Corynebacterium</td>\n",
       "      <td>931089.4</td>\n",
       "      <td>fig|931089.4.peg.69</td>\n",
       "      <td>atgaccattctcgacatcgcacgcctccaagttctggacaacggcg...</td>\n",
       "      <td>MTILDIARLQVLDNGEGLNQKQLMEVLTLPEDQIPDLMELAHQVRL...</td>\n",
       "      <td>dep</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       seqid         evalue  bitscore  pident  \\\n",
       "0    gi|1884111766|ref|NZ_JACDUR010000005.1|  1.470000e-180     631.0  75.362   \n",
       "1       gi|1254547583|ref|NZ_OBDY01000010.1|  1.220000e-175     614.0  74.920   \n",
       "2    gi|1892753580|ref|NZ_JACHIR010000001.1|   0.000000e+00     753.0  77.409   \n",
       "3    gi|1892940858|ref|NZ_JACHGN010000004.1|   0.000000e+00     725.0  76.737   \n",
       "4           gi|1850378389|ref|NZ_AP022870.1|   0.000000e+00     655.0  75.602   \n",
       "..                                       ...            ...       ...     ...   \n",
       "187         gi|2175968702|ref|NZ_CP090063.1|   0.000000e+00     744.0  77.503   \n",
       "188     gi|1697805133|ref|NZ_BJNG01000006.1|   0.000000e+00     866.0  80.144   \n",
       "189     gi|1986697868|ref|NZ_BOOC01000053.1|   0.000000e+00     673.0  75.635   \n",
       "190     gi|1816837986|ref|NZ_BLKW01000004.1|   0.000000e+00    1098.0  84.438   \n",
       "191          gi|924867397|ref|NZ_CP009220.1|  2.010000e-128     456.0  71.109   \n",
       "\n",
       "     length   staxid                             ssname  \\\n",
       "0       966  1032476                    Nonomuraea soli   \n",
       "1       937  1036182        Actinoplanes atraurantiacus   \n",
       "2       965   103725                Kutzneria kofuensis   \n",
       "3       993  1073253  Thermocatellispora tengchongensis   \n",
       "4       955  1076124               Phytohabitans flavus   \n",
       "..      ...      ...                                ...   \n",
       "187     969   715473           Amycolatopsis acidiphila   \n",
       "188     972    76726  Pseudonocardia hydrocarbonoxydans   \n",
       "189     985    83302             Microbispora corallina   \n",
       "190     996    84962            Mycobacterium botniense   \n",
       "191     938   931089  Corynebacterium deserti GIMN1.010   \n",
       "\n",
       "                              scomname   sstart    ssend  ... NCBI Taxon ID  \\\n",
       "0                      Nonomuraea soli  1028404  1029363  ...       1032476   \n",
       "1          Actinoplanes atraurantiacus    77996    77063  ...       1036182   \n",
       "2                  Kutzneria kofuensis  5947414  5948375  ...        103725   \n",
       "3    Thermocatellispora tengchongensis   576796   577782  ...       1073253   \n",
       "4                 Phytohabitans flavus  9487101  9486152  ...       1076124   \n",
       "..                                 ...      ...      ...  ...           ...   \n",
       "187           Amycolatopsis acidiphila  6252312  6251348  ...        715473   \n",
       "188  Pseudonocardia hydrocarbonoxydans    46003    45037  ...         76726   \n",
       "189             Microbispora corallina    20894    21873  ...         83302   \n",
       "190            Mycobacterium botniense  1033100  1034095  ...         84962   \n",
       "191  Corynebacterium deserti GIMN1.010    68759    69694  ...        931089   \n",
       "\n",
       "     Genome ID                Order                Family               Genus  \\\n",
       "0    1032476.3  Streptosporangiales  Streptosporangiaceae          Nonomuraea   \n",
       "1    1036182.3    Micromonosporales    Micromonosporaceae        Actinoplanes   \n",
       "2     103725.4    Pseudonocardiales    Pseudonocardiaceae           Kutzneria   \n",
       "3    1073253.3  Streptosporangiales  Streptosporangiaceae  Thermocatellispora   \n",
       "4    1076124.3    Micromonosporales    Micromonosporaceae       Phytohabitans   \n",
       "..         ...                  ...                   ...                 ...   \n",
       "187   715473.5    Pseudonocardiales    Pseudonocardiaceae       Amycolatopsis   \n",
       "188    76726.4    Pseudonocardiales    Pseudonocardiaceae      Pseudonocardia   \n",
       "189    83302.3  Streptosporangiales  Streptosporangiaceae        Microbispora   \n",
       "190    84962.6    Corynebacteriales      Mycobacteriaceae       Mycobacterium   \n",
       "191   931089.4    Corynebacteriales    Corynebacteriaceae     Corynebacterium   \n",
       "\n",
       "    feature.genome_id       feature.patric_id  \\\n",
       "0           1032476.3  fig|1032476.3.peg.5624   \n",
       "1           1036182.3  fig|1036182.3.peg.2302   \n",
       "2            103725.4   fig|103725.4.peg.5640   \n",
       "3           1073253.3  fig|1073253.3.peg.2482   \n",
       "4           1076124.3  fig|1076124.3.peg.9236   \n",
       "..                ...                     ...   \n",
       "187          715473.5   fig|715473.5.peg.6308   \n",
       "188           76726.4     fig|76726.4.peg.840   \n",
       "189           83302.3    fig|83302.3.peg.7395   \n",
       "190           84962.6    fig|84962.6.peg.3194   \n",
       "191          931089.4     fig|931089.4.peg.69   \n",
       "\n",
       "                                   feature.na_sequence  \\\n",
       "0    gtgacgatcctcgacatcgcccgcacccaggtcctggagcagggca...   \n",
       "1    atgcccgagatcctcgagatggcccgtgcccaggttctcgccggcg...   \n",
       "2    gtgaccgccagcgtggacaccgacatcatcgccgtcgcccgtgagc...   \n",
       "3    gtggacattctgaccgtcgcccgcgaccaggtgctcgaccggggcg...   \n",
       "4    atgccagagatcctcgacatcgctcgtgcccaggtgctcgaacgcg...   \n",
       "..                                                 ...   \n",
       "187  gtgacctcagcacccgaacagaccgacgtggtcgccgtcgcccggg...   \n",
       "188  atggccgaagccgccgccatcctcgccaccgcccgcacccgggtgc...   \n",
       "189  gtgaacactgacatcgtggagatcgcccgcgtccaggtgctcgagg...   \n",
       "190  gtgacgcaggcgatgacccggccggccgccgatgacagcaacaccg...   \n",
       "191  atgaccattctcgacatcgcacgcctccaagttctggacaacggcg...   \n",
       "\n",
       "                                   feature.aa_sequence type  \n",
       "0    MTILDIARTQVLEQGKGLDREQALRCLLLDDALLPDLLALAHEVRM...  ind  \n",
       "1    MPEILEMARAQVLAGGAGLGEAQILEVLRLGDDHLPELLQLAHDVR...  ind  \n",
       "2    MTASVDTDIIAVAREQVLERGEGLSQEQLLQVLKLGDDRLAELLQL...  ind  \n",
       "3    MDILTVARDQVLDRGEGLTAEQALECLRLPDERLPELLALAHEVRV...  ind  \n",
       "4    MPEILDIARAQVLERGTGLDEAGVLAVLRLPDEHVPDALQLAHEVR...  ind  \n",
       "..                                                 ...  ...  \n",
       "187  MTSAPEQTDVVAVAREQVLEQGIGLSQRQVLDVLRLADDRLSELLA...  dep  \n",
       "188  MAEAAAILATARTRVLEEGVGLDEAQVLEVLRLPDEALDDLLQLAH...  dep  \n",
       "189  MNTDIVEIARVQVLEEGRGLDAAQALECLTLPDDRLPELLALAHEV...  dep  \n",
       "190  MTQAMTRPAADDSNTDVLAVARRQVLELGKGLTRDQVLQVLQLPDE...  dep  \n",
       "191  MTILDIARLQVLDNGEGLNQKQLMEVLTLPEDQIPDLMELAHQVRL...  dep  \n",
       "\n",
       "[192 rows x 22 columns]"
      ]
     },
     "execution_count": 292,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "56b141b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "write_df_to_fasta(merged_df, 'feature.aa_sequence', 'Genome Name', \"motif/biob_merged_subset200_filter3.fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "id": "d3530232",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df = merged_df[['Genome Name', 'staxid', 'Order', 'Family', 'Genus', 'type']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "31982136",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add manually added data:\n",
    "\n",
    "data = [\n",
    "    [\"Escherichia coli CFT073\", 199310, \"Enterobacterales\", \"Enterobacteriaceae\", \"Escherichia\", \"ind\"],\n",
    "    [\"Mycobacterium smegmatis MC2155\", 246196, \"Corynebacteriales\", \"Mycobacteriaceae\", \"Mycobacterium\", \"dep\"],   \n",
    "    [\"Mycobacterium tuberculosis H37Rv\", 83332, \"Corynebacteriales\", \"Mycobacteriaceae\", \"Mycobacterium\", \"dep\"]\n",
    "    ]\n",
    "\n",
    "merged_df = merged_df.append(pd.DataFrame(data, columns=merged_df.columns)).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "86aa422d",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df['name'] = merged_df['Genome Name'].str.replace(\" \", \"_\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "id": "04953ebe",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df.to_csv(\"motif/biob_merged_subset200_annotation_filter3.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "801cfaa9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7be743d2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b56c58b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cd9cc05",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "2087a2d7",
   "metadata": {},
   "source": [
    "## read in msa made from merged independent/dependent biob dataset and split into seperate independent and dependent mfa files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "d49ca467",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = read_in_fasta(\"motif/merged_biob_seqs_aa.mfa\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "c447121a",
   "metadata": {},
   "outputs": [],
   "source": [
    "independent_species = ind['id'].tolist()\n",
    "dependent_species = dep['id'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "e53ee2f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ornithinimicrobium_tianjinense_strain_CGMCC_1....</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Sphaerimonospora_thailandensis_strain_NBRC_107569</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Conexibacter_sp._Seoho-28</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Conexibacter_arvalis_strain_DSM_23288</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Nitriliruptor_alkaliphilus_DSM_45188</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>Dactylosporangium_siamense_strain_NBRC_106093</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>Virgisporangium_aliadipatigenens_strain_NBRC_1...</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>Planosporangium_thailandense_strain_TBRC_5610</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>Planosporangium_mesophilum_strain_NBRC_109066</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>Planosporangium_flavigriseum_strain_NBRC_105377</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    id  \\\n",
       "0    Ornithinimicrobium_tianjinense_strain_CGMCC_1....   \n",
       "8    Sphaerimonospora_thailandensis_strain_NBRC_107569   \n",
       "10                           Conexibacter_sp._Seoho-28   \n",
       "11               Conexibacter_arvalis_strain_DSM_23288   \n",
       "12                Nitriliruptor_alkaliphilus_DSM_45188   \n",
       "..                                                 ...   \n",
       "804      Dactylosporangium_siamense_strain_NBRC_106093   \n",
       "805  Virgisporangium_aliadipatigenens_strain_NBRC_1...   \n",
       "806      Planosporangium_thailandense_strain_TBRC_5610   \n",
       "807      Planosporangium_mesophilum_strain_NBRC_109066   \n",
       "808    Planosporangium_flavigriseum_strain_NBRC_105377   \n",
       "\n",
       "                                                   seq  \n",
       "0    (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "8    (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "10   (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "11   (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "12   (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "..                                                 ...  \n",
       "804  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "805  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "806  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "807  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "808  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "\n",
       "[340 rows x 2 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['id'].isin(independent_species)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "b37418b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Nocardia_sp._YIM_PH_21724</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Kribbella_jejuensis_strain_DSM_17305</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Nocardia_sp._ET3-3</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mycobacterium_haemophilum_DSM_44634_ATCC_29548</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Mycobacterium_vulneris_strain_DSM_45247</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>Microbispora_sp._NEAU-HEGS1-5</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611</th>\n",
       "      <td>Microbispora_camponoti_strain_2C-HV3</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>612</th>\n",
       "      <td>Streptosporangium_roseum_DSM_43021</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>Streptosporangium_subroseum_strain_CGMCC_4.2132</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>Sporichthya_polymorpha_DSM_43042</td>\n",
       "      <td>(-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>469 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  id  \\\n",
       "1                          Nocardia_sp._YIM_PH_21724   \n",
       "2               Kribbella_jejuensis_strain_DSM_17305   \n",
       "3                                 Nocardia_sp._ET3-3   \n",
       "4     Mycobacterium_haemophilum_DSM_44634_ATCC_29548   \n",
       "5            Mycobacterium_vulneris_strain_DSM_45247   \n",
       "..                                               ...   \n",
       "610                    Microbispora_sp._NEAU-HEGS1-5   \n",
       "611             Microbispora_camponoti_strain_2C-HV3   \n",
       "612               Streptosporangium_roseum_DSM_43021   \n",
       "614  Streptosporangium_subroseum_strain_CGMCC_4.2132   \n",
       "631                 Sporichthya_polymorpha_DSM_43042   \n",
       "\n",
       "                                                   seq  \n",
       "1    (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "2    (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "3    (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "4    (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "5    (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "..                                                 ...  \n",
       "610  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "611  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "612  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "614  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "631  (-, -, -, -, -, -, -, -, -, -, -, -, -, -, -, ...  \n",
       "\n",
       "[469 rows x 2 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['id'].isin(dependent_species)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "802df898",
   "metadata": {},
   "outputs": [],
   "source": [
    "write_df_to_fasta(df[df['id'].isin(dependent_species)], 'seq', 'id', 'merged_align_biob_dependent_seqs_aa.mfa')\n",
    "write_df_to_fasta(df[df['id'].isin(independent_species)], 'seq', 'id', 'merged_align_biob_independent_seqs_aa.mfa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3dd2f030",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
